---
title: "Transcriptomic level adaptive capacity play roles in the deterministic assembly of a two species cocultures"
author: "Chun-Hui Gao, Hui Cao, Peng Cai"
date: "2020-11-16"
output: bookdown::html_document2
bibliography: references.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(fig.width = 8,fig.asp=0.618,message = F,dpi = 300)

# 作图
library(tidyverse)
library(ggplot2)
library(cowplot)
library(ggpubr)
library(corrplot)
library(pheatmap)
library(RColorBrewer)
library(vegan)
library(DESeq2)
library(openxlsx)
library(clusterProfiler)
theme_set(theme_bw())

# load user defined functions
source("./functions/functions.R")
source("./functions/ggplot_smooth_func.R")
 
```

## Growth of *E. coli* and *P. putida* in monoculture and cocultures

`qPCR_data` contains the result of species-specific quantitative PCR. It has five columns:

-   sample: Sample name
-   condition: monoculture and coculture names. The two-species coculture system has five different initial abundance, two of which were monocultures, and the other three had the initial ratio (EC/PP) of 1:1000, 1:1, 1000:1, respectively. For convenience, we named the *P. putida* monoculture, 1:1000, 1:1, 1000:1 cocultures and *E. coli* monoculture as related to the proportion of *E. coli* to five groups of "none", "less", "equal", "more" and "all", respectively.
-   time: sample time.
-   species: species name. In cocultures, the quantity was related to two species, respectively.
-   quantity: the quantity of species, as calculated from the qPCR CT value and standard curves.

```{r qPCR-data}
qPCR_data <- read.csv("data/qPCR-data.csv")
qPCR_data$condition <- factor(qPCR_data$condition, 
                              levels = c("none","less","equal","more","all"))

head(qPCR_data)

organism <- c("EC","PP")
organsim_fullname <- c("EC"="E. coli","PP"="P. putida")
```

Figure \@ref(fig:growth-curve) shows Growth curves of *E. coli* and *P. putida* in monoculture (A) and the "1:1000", "1:1", "1000:1" cocultures (B-D). The quantities were determined using species-specific qPCR as described in methods. In B-D subplots, the growth curves of monocultures were placed on the background layer (dashed lines), showing the difference between monoculture and cocultures.

```{r growth-curve, fig.asp=0.8, fig.cap="Growth curves of *E. coli* and *P. putida* in monoculture (A) and the 1:1000, 1:1, 1000:1 cocultures (B-D)"}

# stats
growth_data <- qPCR_data %>% group_by(species,time,condition)  %>%
  summarise(y=median(log10(quantity)),std=sd(log10(quantity),na.rm = T))

# separate monoculture and coculture
monoculture <-  growth_data %>%
  filter(condition %in% c("none","all"))
coculture <- growth_data %>%
  filter(condition %in% c("less","equal","more"))

# plot
growth_curve_mono <- ggplot(monoculture,aes(time,y,color=species)) +
  geom_line(lty="dashed",size=1) +
    geom_errorbar(aes(ymin=y-std,ymax=y+std),size=.5) + 
    geom_point(data = monoculture) 
growth_curves_coculture <- lapply(c("less","equal","more"), function(x){
  ggplot(mapping = aes(time,y,color=species)) + 
    geom_line(lty="dashed",size=1,alpha=1/3,data = monoculture) +
    geom_line(size=1,data=filter(coculture,condition==x)) +
    geom_errorbar(aes(ymin=y-std,ymax=y+std),size=.5,data = monoculture,alpha=1/3) + 
    geom_errorbar(aes(ymin=y-std,ymax=y+std),size=.5,data = filter(coculture,condition==x)) + 
    geom_point(data = monoculture,alpha=1/3) + 
    geom_point(data=filter(coculture,condition==x))
    
})

# plot enhancement
growth_curves <- c(list(growth_curve_mono), growth_curves_coculture)
growth_curves <- lapply(growth_curves, function(p){
  p + ylab("Log(quantity)") + xlab("Time (h)") +
        scale_y_continuous(limits = c(5,10.8),breaks = 5:10) +
        scale_color_discrete(labels=organsim_fullname,name="Species") +
        theme_bw() +
        theme(legend.position = c(0.7,0.2),
              legend.text = element_text(face = "italic"))
})

# show plot
plot_grid(plotlist = growth_curves,ncol=2,labels = "AUTO")
# ggsave("figure 1.tiff",path="figures")
# export::graph2ppt(file="figures.pptx",append=TRUE)

```

### Comparision of coculture and monoculture quantities

Figure \@ref(fig:monoculture-vs-coculture) shows the variance analysis of the species abundances between monoculture and three cocultures. Results for *E. coli* (A-G) and *P. putida* (H-N) were given by time, as been illustrated on the top subtitle. In x-axis, mono represents the quantity of monoculture, and "1:1000", "1:1", "1000:1" represent the quantity of three cocultures. The significance of p-values were showed by "\*" (p\<0.05), "\*\*" (p\<0.01) or "ns" (p\>0.05).

```{r monoculture-vs-coculture, fig.width=12,fig.cap="Analysis the variance of the species abundances between monoculture and three cocultures"}
ref_group <- c("all","none")
library(ggtext)
plots <- lapply(seq_along(ref_group), function(i){
  ref <- ref_group[[i]]
  lapply(c(0,0.5,1,2,4,8,24), function(x){
   df <- qPCR_data %>% 
     filter(species == organism[[i]],time==x) %>% 
     dplyr::select(condition,quantity)
  df$condition <- relevel(df$condition, ref)
  ggplot(df,aes(condition,log10(quantity))) + 
    geom_boxplot() + 
    geom_jitter() + 
    scale_x_discrete(labels = c("mono","1:1000","1:1","1000:1")) +
    scale_y_continuous(expand = c(0,1)) +
    labs(subtitle = paste0("*",organsim_fullname[[organism[[i]]]],"* - ", x, "h")) +
    stat_compare_means(ref.group = ref,label = "p.signif") +
    theme(plot.subtitle = element_markdown()) 
  })
  
})
plots <- unlist(plots,recursive = F)
plot_grid(plotlist = plots,ncol=5,labels = "AUTO")

```

### Ratio (EC/PP) changes in cocultures

Ratios of EC/PP in cocultures were calculated, and ratio changes in cocultures were compared in Figure \@ref(fig:ratio-change).

```{r}
# calculate ratio in cocultures
ratio <- qPCR_data %>% 
  filter(condition %in% c("less","equal","more")) %>% 
  group_by(sample, condition, species, time) %>% 
  mutate(rep = row_number()) %>% 
  ungroup() %>% 
  pivot_wider(names_from = species, values_from = quantity) %>%
  mutate(ratio = EC/PP) %>%
  filter(!is.na(ratio))
```

Figure \@ref(fig:ratio-change) shows the deterministic assembly of *E. coli* and *P. putida* cocultures. (A) the real-time ratio of EC: PP after 0, 0.5, 1, 2, 4, 8 and 24 h cultivation. (B) Analysis of the variances of EC/PP ratios (24h) between cocultures.

```{r ratio-change, fig.width=6, fig.asp=0.618}
ratio.sum <- ratio %>% 
  group_by(sample) %>%
  mutate(y=mean(ratio,na.rm=TRUE),std=sd(ratio,na.rm=TRUE)) %>% 
  dplyr::select(sample,condition, time, y, std) %>%
  unique()
ratio.sum$condition <- factor(ratio.sum$condition,
                        levels = c("less","equal","more"),
                        labels = c("1:1000","1:1","1000:1")) 
plot_ratio <- ggplot(ratio.sum, aes(time,y,shape=condition,color=condition)) + 
  geom_rect(aes(xmin=23,xmax=25,ymin=0.02,ymax=0.3),
            fill="lightyellow",color="black",alpha=0.1) +
  geom_line(size=1,show.legend = F) +
  geom_point(size=2,show.legend = F) +
  geom_errorbar(aes(ymin=y-std,ymax=y+std),show.legend = F) +
  geom_text(aes(x=9,label=condition),hjust=0,vjust=c(0,0,1),
            data = filter(ratio.sum,time==8),
            show.legend = F) +

  # directlabels::geom_dl(aes(label=condition),method="smart.grid") +
  scale_y_log10(labels=formatC,breaks=10^(-3:3)) +
  labs(x="Time (h)", y="Ratio (EC/PP)") +
  # scale_x_continuous(limits = c(-5,NA)) +
  theme(legend.position = c(0.8,0.75))

ratio_24h <- ratio %>% filter(time==24)
plot_ratio_stats <- ggplot(ratio_24h,aes(condition,ratio,color=condition)) +
  geom_boxplot(fill="lightyellow") + 
  geom_jitter() + 
  stat_compare_means(
    comparisons = list(c("less","equal"),c("less","more"),c("equal","more")),
    label="p.format") +
   scale_x_discrete(breaks=c("less","equal","more"),
                    labels=c("1:1000","1:1","1000:1"))+
  xlab("Condition") + ylab("Ratio (EC/PP)") +
  theme(legend.position = "none",
        panel.background = element_rect(fill="lightyellow"
      ))

plot_grid(plot_ratio,plot_ratio_stats,labels = "AUTO",rel_widths = c(3,2))
# ggsave("figure 2.tiff",path="figures")
# export::graph2ppt(file="figures.pptx",append=TRUE)
```

Notably, the ratio differences were non-significant by time in the logarithmic phase, but were all significant in the stationary phase for every coculture (Figure \@ref(fig:comparison-by-time)).

For each coculture, significance of variances between each sample were as follows:

-   the 1:1000 coculture

```{r}
### stats
ratio1 <- ratio %>% filter(condition=="less")
(p1 <- pairwise.wilcox.test(ratio1$ratio,ratio1$time,p.adjust.method = "BH"))
```

-   the 1:1 coculture

```{r}
ratio2 <- ratio %>% filter(condition=="equal")
(p2 <- pairwise.wilcox.test(ratio2$ratio,ratio2$time,p.adjust.method = "BH"))
```

-   and the 1000:1 coculture

```{r}
ratio3 <- ratio %>% filter(condition=="more")
(p3 <- pairwise.wilcox.test(ratio3$ratio,ratio3$time,p.adjust.method = "BH"))
```

Figure \@ref(fig:comparison-by-time) shows P-values of pairwise comparison of real-time EC/PP ratios in the "1:1000" (A), "1:1" (B) and "1000:1" (C) cocultures. Circles showed the adjusted p-value for the comparison of the EC/PP ratios between two samples, which were indicated on the top and left (large circle mean large p-value). On this plot, a cross mark was given if the p-value is non-significant (P \> 0.05).

```{r comparison-by-time, fig.cap="P-values of pairwise comparison of real-time EC/PP ratios in the “1:1000” (A), “1:1” (B) and “1000:1” (C) cocultures.", fig.asp=0.4}
par(mfrow=c(1,3))
pvalues <- list(p1,p2,p3)
line = 0
cex = 1.2
side = 3
adj=0.15
plots <- lapply(seq_along(pvalues), function(i){
  x <- pvalues[[i]]
  corrplot(x$p.value, 
           type = "lower",
           col = "grey",
           cl.pos = "n",
           is.corr = FALSE, 
           method = "circle",
           p.mat = x$p.value,
           sig.level = 0.05)
  mtext(LETTERS[[i]], side = side, line = line, cex = cex, adj = adj)

})

```

## Gene expression analysis

To reveal the mechanism of community assembly in cocultures, we investigated the transcriptomic changes in cocultures using RNA-seq analysis.

Totally 60 samples were sequences. After sequencing quality control, each sample has 2.6 -- 3.9 M paired reads, and 3.9 -- 5.9 G base pairs, having an overall coverage of 300 X at least. After filtration, high-quality reads were aligned against the P. putida (<https://www.ncbi.nlm.nih.gov/genome/?term=pseudomonas+putida+kt2440>) and E.coli reference genome (<https://www.ncbi.nlm.nih.gov/genome/?term=Escherichia+coli+K-12>) using hisat2 and gene expression changes were quantified using DESeq2 software [@herbergDiagnosticTestAccuracy2016; @loveModeratedEstimationFold2014]. While aligning to reference genomes, the overall aligned rates are ranging from 97.23% to 98.43%.

```{r}
tableS1 <- read.xlsx("./data/tableS1.xlsx")
summary(tableS1)
```

Figure \@ref(fig:reads-count) shows the reads count of *E. coli* and *P. putida* in each RNA-seq library. Samples were taken with each condition at indicated times (at 0, 4, 8 and 24h). (A-D) *P. putida* monoculture samples; (E-H) *E. coli* and *P. putida* “1:1000” coculture; (I-L) the “1:1” coculture; (M-P) the “1000:1” coculture; and (Q-T) the *E. coli* monoculture samples. Each sample has three replicates. Only aligned reads to either *E. coli* or *P. putida* genome were used in this calculation. Plots showed the proportion of reads which have aligned to corresponding genome by different colors, as indicated on the top.

```{r reads-count, fig.asp=1.5,width=4.5, fig.cap="Reads count of *E. coli* and *P. putida* in each RNA-seq library"}
ht_counts <- readRDS(file = "./data/ht_counts.rds")
ht_counts$group <- factor(ht_counts$group,
                          levels = c("none_0h","none_4h","none_8h","none_24h","less_0h","less_4h","less_8h","less_24h","equal_0h","equal_4h","equal_8h","equal_24h","more_0h","more_4h","more_8h","more_24h","all_0h","all_4h","all_8h","all_24h"),
                          labels = c("P. putida_0h","P. putida_4h","P. putida_8h","P. putida_24h","1:1000_0h","1:1000_4h","1:1000_8h","1:1000_24h","1:1_0h","1:1_4h","1:1_8h","1:1_24h","1000:1_0h","1000:1_4h","1000:1_8h","1000:1_24h","E. coli_0h","E. coli_4h","E. coli_8h","E. coli_24h"))
ht_counts_total <- ht_counts %>% group_by(sample_id, group, organism) %>%
  summarise(sum_of_reads=sum(count)) %>%
  group_by(sample_id) %>% 
  mutate(proportion=sum_of_reads/sum(sum_of_reads))
samples <- levels(ht_counts_total$group)
plots <- lapply(1:length(samples),function(i){
  sample_group <- samples[[i]]
  df <- filter(ht_counts_total,group==sample_group)
  ggplot(df,aes(x=sample_id, y=proportion,fill=organism)) + 
    geom_bar(stat = "identity",position = "stack") + 
    scale_y_continuous(labels = function(l)paste(format(l*100,digits = 2),"%",sep="")) +
    scale_x_discrete(labels=c("Rep1","Rep2","Rep3")) +
    scale_fill_discrete(name="Organism: ",labels=c("EC"="E. coli","PP"="P. putida")) +
    labs(title = sample_group) +
    theme(legend.text = element_text(face = "italic"),
          legend.position = "none",
          axis.title = element_blank())
})
legend <- get_legend(plots[[1]] + theme(legend.position = "top",legend.direction = "horizontal"))
plot_grid(legend, plot_grid(plotlist = plots,labels = "AUTO",ncol=4),rel_heights = c(1,15),ncol=1)
# ggsave("figure S3.tiff",path="figures")
# export::graph2ppt(file="figures.pptx",append=TRUE)

```

### Identify gene expression changes

This step is a time consuming step. Use the precalculated DEG if possible.

load precalculated DEG results.

```{r}
dds.EC <- readRDS("data/dds.EC.2.rds")
dds.PP <- readRDS("data/dds.PP.2.rds")
DEG_results.EC <- readRDS("data/DEG_results.EC.rds")
DEG_results.PP <- readRDS("data/DEG_results.PP.rds")
```

### RNA-seq clustering

```{r}
list_of_vsd <- lapply(list(dds.EC,dds.PP),function(dds){
  vst(dds,blind = F)
})
list_of_vsd[[1]]$ratio0 <- factor(list_of_vsd[[1]]$ratio0,
                                  levels = c("none","less","equal","more","all"),
                        labels = c("P. putida","1:1000","1:1","1000:1","E. coli"))
list_of_vsd[[2]]$ratio0 <- factor(list_of_vsd[[2]]$ratio0,
                                  levels = c("none","less","equal","more","all"),
                        labels = c("P. putida","1:1000","1:1","1000:1","E. coli"))

```

```{r RNA-seq-PCA, fig.width=10,fig.asp=0.618,fig.cap="Principle coordinate analysis (PCA) of the gene expression profiles of *E. coli* (A) and *P. putida* (B) in different samples."}
list_of_PCA_plot <- lapply(list_of_vsd, function(vsd) {
  myPlotPCA(vsd) + 
    facet_wrap(~time,ncol=4) +
    directlabels::geom_dl(aes(label=ratio0),method = "smart.grid",size=2) +  #文本代替标签 位置标注的不好,改size没用
    scale_color_manual(limits=c("E. coli","1:1000","1:1","1000:1","P. putida"),
                       values = brewer.pal(5,"Dark2")) +
    theme(legend.position = "none")
  })

plot_grid(plotlist = list_of_PCA_plot,labels = "AUTO",ncol=1)
# ggsave("figure 3.tiff",path="figures")
# export::graph2ppt(x=plot_grid(plotlist = list_of_PCA_plot,labels = "AUTO",ncol=1),file="PCA.pptx",width=6,height=4)

```
### Count of DEGs

Figure \@ref(fig:count-of-deg) shows the Counts of differentially expressed genes (DEGs) in three cocultures, in *E. coli* (A) and *P. putida* (B). The DEGs were identified by comparison with the corresponding monoculture at the same time. Up- and down-regulation of genes were colored by red and cyan, respectively.

```{r count-of-deg, fig.asp=0.618,fig.cap="Counts of differentially expressed genes (DEGs) in three cocultures, in *E. coli* (A) and *P. putida* (B).",fig.width=6}
## count DEG
deg_count <- function(data){
  do.call("rbind",lapply(data, function(x) table(x$expression))) %>%
    as.data.frame() %>%
    rownames_to_column(var = "name") %>%
    separate(name, into = c("ratio","time"), sep = "_", extra = "drop") %>%
    mutate(time = as.numeric(str_extract(time, "[0-9]+"))) %>%
    pivot_longer(cols = c("dn","up"), names_to = "type", values_to = "count") %>%
    mutate(count = ifelse(type == "dn", -count, count)) %>%
    complete(ratio, time, type, fill = list(count = 0)) %>%
    mutate(ratio = factor(ratio, 
                          levels = c("less","equal","more"),
                          labels = c("1:1000","1:1","1000:1")),
           type = factor(type,
                         levels = c("up","dn"),
                         labels = c("Up","Down")))
}


deg_count_EC <- deg_count(DEG_results.EC)
deg_count_PP <- deg_count(DEG_results.PP)
count <- list(deg_count_EC, deg_count_PP)
plots <- lapply(seq_along(count), function(i){
  x <- count[[i]]
  ggplot(x, aes(x = time, y = count, color = type)) +
    geom_point() +
    geom_line(size = 1) +
    scale_y_continuous(labels = function(x){abs(x)}) +
    facet_wrap(~ratio) +
    labs(x="Time(h)",y="Count of DEGs",
         color = "Gene expression:",
         title= paste0("DEGs in *",organsim_fullname[[i]],"*")) +
    theme(legend.position = c(0.618,1),
        legend.justification = c(0.5,-0.65),
        legend.direction = "horizontal",
        plot.title = element_markdown())
}) 


plot_grid(plotlist = plots,labels = "AUTO",ncol = 1)
# ggsave("figure 4.tiff", path="figures")
# export::graph2ppt(file="figures.pptx",append=TRUE)
```

### Specific DEGs in *E. coli* and *P. putida*

Figure \@ref(fig:deg-venn) shows the comparision of differentially expressed genes by time in *E. coli* (A-C) and *P. putida* (D-F). Although DEGs overlapped for different time, specific genes are the majority in almost every time point. For instance, 57 out of 93 DEGs are specific in 1:1000 coculture in *E. coli* (Fig. \@ref(fig:deg-venn)A).

```{r deg-venn, fig.width=8, fig.cap="Comparision of differentially expressed genes by time"}
library(ggVennDiagram)

ratio0 <- c("1:1000","1:1","1000:1")
deg_Venn_plot_EC <- lapply(seq_along(ratio0), function(i){
  gene_list <- lapply(DEG_results.EC[(i*4-3):(i*4)], function(x){x$gene})
  ggVennDiagram(gene_list,label = "count",
                category.names = c("0h","4h","8h","24h")) +
    scale_fill_gradient(low="white",high="red",limits=c(0,310)) +
    coord_equal() +
    labs(title=paste0(ratio0[[i]]," - *E. coli*")) +
    theme(legend.position = "none",
          plot.title = element_markdown(hjust=0.5))
})

deg_Venn_plot_PP <- lapply(seq_along(ratio0), function(i){
  gene_list <- lapply(DEG_results.PP[(i*4-3):(i*4)], function(x){x$gene})
  ggVennDiagram(gene_list,label = "count",
                category.names = c("0h","4h","8h","24h")) +
    scale_fill_gradient(low="white",high="red",limits=c(0,310)) +
    coord_equal() +
    labs(title=paste0(ratio0[[i]]," - *P. putida*")) +
    theme(legend.position = "none",
          plot.title = element_markdown(hjust=0.5))
})

plot_grid(plotlist = c(deg_Venn_plot_EC,deg_Venn_plot_PP),
          labels = "AUTO")
# ggsave("figure S9.tiff",path="figures")
```

## Enrichment analysis of DEGs

We use `clusterProfiler` to perform KEGG enrichment analysis.

```{r}
ck_plot <- function(ck){
  df <- data.frame(ck) %>%
    mutate(ratio = factor(ratio, levels = c("less","equal","more"),
                          labels = c("1:1000","1:1","1000:1")),
           time = factor(time, levels = c("0h","4h","8h","24h")),
           expression = factor(expression, 
                               levels = c("up","dn"),
                               labels = c("up-regulated", "down-regulated"))) 
  df$Description <- factor(df$Description, levels = kegg_path_tree(df,pathway_name = "Description", gene_id = "geneID"))
  df$GeneRatio <- sapply(df$GeneRatio, function(x) eval(parse(text = x)))
  ggplot(df, aes(time, Description, size = GeneRatio,color=expression)) +
    geom_point() +
    facet_grid(~ ratio, scales = "free_y") +
    labs(y="KEGG pathway")
}


grid_panel_autoheight <- function(p){
  # 调整panel的高度
  require(gtable)
  gp <- ggplotGrob(p)
  # gtable_show_layout(gp)
  facet.rows <- gp$layout$t[grepl("panel",gp$layout$name)]
  y.var <- sapply(ggplot_build(p)$layout$panel_scales_y,
                  function(l) length(l$range$range))
  gp$heights[facet.rows] <- gp$heights[facet.rows] * y.var
  return(gp)
}

reset_facet_width <- function(p, rel_width = c(1)){
  gp <- ggplotGrob(p)
  facet.cols <- gp$layout$l[grepl("panel",gp$layout$name)]
  gp$widths[facet.cols] <- gp$widths[facet.cols] * rel_width
  return(gp)
}

kegg_path_tree <- function(df, pathway_name, gene_id, sep="/"){
  pathway <- df %>% 
    separate_rows(gene_id, sep = sep) %>%
    dplyr::select(c(pathway_name, gene_id)) %>%
    unique() %>%
    mutate(value = 1) %>%
    pivot_wider(id_cols = pathway_name,
                names_from = gene_id,
                values_from = value,
                values_fill = list(value = 0))
  library(vegan)
  matrix <- as.matrix(column_to_rownames(pathway, pathway_name))
  dist <- vegdist(matrix, method = "jaccard")
  
  library(ape)
  library(ggtree)
  tree <- bionj(dist)
  p <- ggtree(tree,branch.length = "none") 
  p$data %>%
    filter(isTip) %>%
    arrange(y) %>%
    pull(label)
}
```


```{r include=FALSE}
## KEGG enrichment in E. coli
deg1 <- do.call("rbind", DEG_results.EC) %>% 
  separate(comparison, into = c("ratio","time"), extra = "drop")

ck1 <- compareCluster(gene ~ ratio + time + expression, 
                      data = deg1, 
                      fun = "enrichKEGG", 
                      organism = "eco") 

p1 <- ck_plot(ck1)
p1 <- grid_panel_autoheight(p1)

## KEGG enrichment in P. putida
deg2 <- do.call("rbind", DEG_results.PP) %>% 
  separate(comparison, into = c("ratio","time"), extra = "drop")

ck2 <- compareCluster(gene ~ ratio + time + expression, 
                      data = deg2, 
                      fun = "enrichKEGG", 
                      organism = "ppu") 

p2 <- ck_plot(ck2)
p2 <- grid_panel_autoheight(p2)
```

(ref:kegg-ora-cap) A dot plot shows the KEGG enrichment result of *E. coli* (A) and *P. putida* (B) in the three coculture as a function of time.

```{r kegg-ora, fig.width=8, fig.asp=1.2, fig.cap="(ref:kegg-ora-cap) "}
plot_grid(p1,p2, rel_heights = c(1,0.4), ncol = 1, labels = "AUTO",align = "v")
```

## Get set enrichment analysis (GSEA)

GSEA use all genes' expressions, so we need to get a unfiltered gene expression data.

```{r}
gene_expression.EC <- readRDS("data/gene_expression.EC.rds")
gene_expression.PP <- readRDS("data/gene_expression.PP.rds")
```



```{r gseKEGG-result-EC, warning=FALSE}
get_genelist <- function(x){
  if (nrow(x) < 1) return(NULL)
  geneList <- x$log2FoldChange
  names(geneList) <- x$gene
  geneList <- sort(geneList,decreasing = T) 
  return(geneList)
}

### *E. coli* GSEA KEGG result
gseKEGG_results.EC <- lapply(gene_expression.EC, function(x){
  geneList <- get_genelist(x)
  tryCatch(gseKEGG(geneList, organism = "eco",nPerm = 10000,minGSSize = 30),error=function(e) NULL)
})

### *P. putida* GSEA KEGG result
gseKEGG_results.PP <- lapply(gene_expression.PP, function(x){
  geneList <- get_genelist(x)
  tryCatch(gseKEGG(geneList, organism = "ppu",nPerm = 10000,minGSSize = 30),error=function(e) NULL)
})
```

```{r}
# combine GSEA result
gse_result <- function(result){
  name <- names(result)
  l <- lapply(seq_along(result), function(i){
    data.frame(result[[i]]) %>%
      mutate(comparison = name[[i]])
  })
  do.call("rbind", l) %>%
    separate(comparison, into = c("ratio","time"), extra = "drop") %>%
    mutate(ratio = factor(ratio, levels = c("less","equal","more"),
                          labels = c("1:1000","1:1","1000:1")),
           time = factor(time, levels = c("0h","4h","8h","24h")),
           type = ifelse(enrichmentScore >0, "activated", "suppressed"),
           enrichScore = abs(enrichmentScore)) 
}

# dotplot GSEA result
gse_dotplot <- function(df){
  ggplot(df, aes(time, Description, size = enrichScore,color=type)) +
    geom_point() +
    facet_grid(~ ratio, scales = "free_y") +
    labs(y="KEGG pathway")
}
```

```{r include=FALSE}
## GSEA dotplot of E. coli
df1 <- gse_result(gseKEGG_results.EC)
df1$Description <- factor(df1$Description,
                          levels = kegg_path_tree(df1,
                                                  pathway_name = "Description",
                                                  gene_id = "core_enrichment"))
p1 <- gse_dotplot(df1)

## GSEA dotplot of P. putida
df2 <- gse_result(gseKEGG_results.PP)
df2$Description <- factor(df2$Description,
                          levels = kegg_path_tree(df2,
                                                  pathway_name = "Description",
                                                  gene_id = "core_enrichment"))
p2 <- gse_dotplot(df2)
```


```{r fig.asp=1.2}
plot_grid(p1,p2,align = "v", ncol = 1,labels = "AUTO")
```








## 带标记的火山图

```{r}
library(DESeq2)
library(tidyverse)


enriched_genes.EC <- lapply(gseKEGG_results.EC,function(x){
  # separate_rows(data.frame(x),"core_enrichment",sep="/")
  df <- data.frame(x)
  if (nrow(df) > 0) {
    separate_rows(df,"core_enrichment",sep="/") %>%
      rename(gene ="core_enrichment")
  }
  else{
    return(NULL)
  }
})
```

```{r fig.width=8,fig.asp=1}
idx <- 1:4
pathways <- unique(do.call("rbind",enriched_genes.EC[idx])$Description)
# color value
color_values <- ggsci::pal_npg("nrc")(length(pathways))
names(color_values) <- pathways
volcano_plots.EC <- lapply(idx, function(i){
  result.raw <- gene_expression.EC[[i]]
  result <- DEG_results.EC[[i]]
  genes <- enriched_genes.EC[[i]]
ggplot() + aes(log2FoldChange,-log10(padj)) +
    geom_point(data = result.raw,color="grey") +
    geom_point(data = result, color="black") +
    geom_point(data=left_join(genes,result),mapping = aes(color=Description)) +
    scale_color_manual(values = color_values) +  
    xlab("Log2Foldchange")+
    theme(legend.position = c(0.01,0.99),
          legend.justification = c(0,1),
          legend.title = element_blank(),
          legend.background = element_blank(),
          legend.box.background = element_rect(color = "grey",fill = NA))
  
})
plot_grid(plotlist = volcano_plots.EC,labels = "AUTO",align = "hv")
# ggsave("figure 5.tiff",path="figures")

```

*P. putida*中没有enriched pathway，只能绘制一个黑白配的火山图。

```{r fig.width=8,fig.asp=1}
# PP in "more" coculture, time 0, 4, 8h
idx <- 9:11
volcano_plots.PP <- lapply(idx, function(i){
  result.raw <- gene_expression.PP[[i]]
  result <- DEG_results.PP[[i]]
  ggplot() + aes(log2FoldChange,-log10(padj)) +
    geom_point(data = result.raw,color="grey") +
    geom_point(data = result, color="black") +
     xlab("Log2Foldchange")
})
plot_grid(plotlist = volcano_plots.PP,labels = "AUTO",align = "hv")
# ggsave("figure S4.tiff",path="figures")

```

# pathway viewer

```{r}
library(pathview)
pathway_ids.EC <- df1$ID
pathway_geneList_names.EC <- df1$core_enrichment
for (i in seq_along(pathway_ids.EC[1:3])){
  idx <- pathway_geneList_names.EC[[i]]
  pathview(gene.data = pathway_geneList_names.EC[[i]],
           pathway.id = pathway_ids.EC[[i]],
           species="eco",
           gene.idtype = "kegg",
           kegg.native = T)

}
```
