<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Chun-Hui Gao, Hui Cao, Peng Cai, et.al." />

<meta name="date" content="2021-07-23" />

<title>Emergent transcriptional adaption facilitates the convergent succession within a synthetic community</title>

<script src="gene-expression_files/header-attrs-2.9/header-attrs.js"></script>
<script src="gene-expression_files/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="gene-expression_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="gene-expression_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="gene-expression_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="gene-expression_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="gene-expression_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="gene-expression_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="gene-expression_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="gene-expression_files/navigation-1.1/tabsets.js"></script>
<link href="gene-expression_files/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="gene-expression_files/highlightjs-9.12.0/highlight.js"></script>
<script src="gene-expression_files/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="gene-expression_files/plotly-binding-4.9.4.1/plotly.js"></script>
<script src="gene-expression_files/typedarray-0.1/typedarray.min.js"></script>
<link href="gene-expression_files/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="gene-expression_files/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="gene-expression_files/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="gene-expression_files/plotly-main-1.57.1/plotly-latest.min.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>




<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div id="header">



<h1 class="title toc-ignore">Emergent transcriptional adaption facilitates the convergent succession within a synthetic community</h1>
<h4 class="author">Chun-Hui Gao, Hui Cao, Peng Cai, et.al.</h4>
<h4 class="date">2021-07-23</h4>

</div>


<div id="growth-of-e.-coli-and-p.-putida-in-monoculture-and-cocultures" class="section level2">
<h2>Growth of <em>E. coli</em> and <em>P. putida</em> in monoculture and cocultures</h2>
<p><code>qPCR_data</code> contains the result of species-specific quantitative PCR. It has five columns:</p>
<ul>
<li>sample: Sample name</li>
<li>condition: monoculture and coculture names. The two-species coculture system has five different initial abundance, two of which were monocultures, and the other three had the initial ratio (EC/PP) of 1:1000, 1:1, 1000:1, respectively. For convenience, we named the <em>P. putida</em> monoculture, 1:1000, 1:1, 1000:1 cocultures and <em>E. coli</em> monoculture as related to the proportion of <em>E. coli</em> to five groups of “none,” “less,” “equal,” “more” and “all,” respectively.</li>
<li>time: sample time.</li>
<li>species: species name. In cocultures, the quantity was related to two species, respectively.</li>
<li>quantity: the quantity of species, as calculated from the qPCR CT value and standard curves.</li>
</ul>
<p>For example:</p>
<pre class="r"><code>qPCR_data &lt;- read.csv(&quot;data/qPCR-data.csv&quot;)
qPCR_data$condition &lt;- factor(qPCR_data$condition, 
                              levels = c(&quot;none&quot;,&quot;less&quot;,&quot;equal&quot;,&quot;more&quot;,&quot;all&quot;))

head(qPCR_data)</code></pre>
<pre><code>##     sample condition time species  quantity
## 1 Sample 1       all    0      EC 150753360
## 2 Sample 1       all    0      EC 179451379
## 3 Sample 1       all    0      EC 197309047
## 4 Sample 1       all    0      EC 325275685
## 5 Sample 1       all    0      EC 389286192
## 6 Sample 1       all    0      EC 299844026</code></pre>
<pre class="r"><code>organism &lt;- c(&quot;EC&quot;,&quot;PP&quot;)
organsim_fullname &lt;- c(&quot;EC&quot;=&quot;E. coli&quot;,&quot;PP&quot;=&quot;P. putida&quot;)</code></pre>
<div id="comparision-of-coculture-and-monoculture-quantities" class="section level3">
<h3>Comparision of coculture and monoculture quantities</h3>
<p>Figure <a href="#fig:monoculture-vs-coculture">1</a> shows Growth curves of <em>E. coli</em> and <em>P. putida</em> in monoculture (A) and the “1:1000,” “1:1,” “1000:1” cocultures (B-D). The quantities were determined using species-specific qPCR as described in methods. In B-D subplots, the growth curves of monocultures were placed on the background layer (dashed lines), shows the variance analysis of the species abundances between monoculture and the “1:1000” (b), “1:1” (d), and “1000:1” (d) cocultures. The significance of p-values were showed by “*” (p&lt;0.05), “**” (p&lt;0.01) or “ns” (p&gt;0.05).</p>
<pre class="r"><code># stats
growth_data &lt;- qPCR_data %&gt;% group_by(species,time,condition)  %&gt;%
  summarise(y=median(log10(quantity)),std=sd(log10(quantity),na.rm = T))

# separate monoculture and coculture
monoculture &lt;-  growth_data %&gt;%
  filter(condition %in% c(&quot;none&quot;,&quot;all&quot;))
coculture &lt;- growth_data %&gt;%
  filter(condition %in% c(&quot;less&quot;,&quot;equal&quot;,&quot;more&quot;))


# plot
growth_curve_mono &lt;- ggplot(monoculture,aes(time,y,color=species)) +
  geom_line(lty=&quot;dashed&quot;,size=1) +
    geom_errorbar(aes(ymin=y-std,ymax=y+std),size=.5) + 
    geom_point(data = monoculture)</code></pre>
<pre class="r"><code>library(rstatix)

qPCR_stat &lt;- qPCR_data %&gt;% mutate(
  condition = fct_other(condition,
                        keep=c(&quot;less&quot;,&quot;equal&quot;,&quot;more&quot;),
                        other_level =  &quot;mono&quot;)) %&gt;%
  group_by(time,species) %&gt;%
  wilcox_test(.,quantity ~ condition,ref.group = &quot;mono&quot;) %&gt;% 
  mutate(condition = group2) %&gt;%
  select(time,species,condition,p.adj,p.adj.signif)


# significance
variance &lt;- qPCR_data %&gt;%
  filter(condition %in% c(&quot;less&quot;,&quot;equal&quot;,&quot;more&quot;)) %&gt;%
  group_by(time,condition,species) %&gt;%
  summarise(y=max(log10(quantity))) %&gt;%
  ungroup() %&gt;%
  left_join(qPCR_stat)

variance_plot &lt;- function(cond){
  qPCR_stat %&gt;% filter(condition == cond) %&gt;%
    left_join(filter(coculture, condition==cond)) %&gt;%
  ggplot(aes(time,y,color=species)) +
  geom_line(size=1,alpha=1/2) +
  geom_point(size=2,alpha=1/2) +
        geom_line(lty=&quot;dashed&quot;,size=1,alpha=1/3,data = monoculture) +
    geom_errorbar(aes(ymin=y-std,ymax=y+std),size=.5,data = monoculture,alpha=1/3) + 
  ggrepel::geom_text_repel(aes(label=p.adj.signif),
                           max.overlaps = 20,
                           segment.colour = &quot;black&quot;,
                           show.legend = F)
}

plots &lt;- lapply(c(&quot;less&quot;,&quot;equal&quot;,&quot;more&quot;), variance_plot)
plots = lapply(c(list(growth_curve_mono),plots), function(p){
  p + 
    ylab(&quot;Log(quantity)&quot;) + xlab(&quot;Time (h)&quot;) +
        scale_y_continuous(limits = c(5,11),breaks = 5:10) +
        scale_color_discrete(labels=organsim_fullname,name=&quot;Species&quot;) +
        theme_bw() +
        theme(legend.position = c(0.7,0.3),
              legend.text = element_text(face = &quot;italic&quot;))
})
plot_grid(plotlist = plots,labels = &quot;auto&quot;, ncol = 3)</code></pre>
<div class="figure" style="text-align: center"><span id="fig:monoculture-vs-coculture"></span>
<img src="gene-expression_files/figure-html/monoculture-vs-coculture-1.png" alt="Analysis the variance of the species abundances between monoculture and cocultures" width="75%" />
<p class="caption">
Figure 1: Analysis the variance of the species abundances between monoculture and cocultures
</p>
</div>
<pre class="r"><code>export::graph2ppt(append  = TRUE)</code></pre>
</div>
<div id="ratio-ecpp-changes-in-cocultures" class="section level3">
<h3>Ratio (EC/PP) changes in cocultures</h3>
<p>Ratios of EC/PP in cocultures were calculated, and ratio changes in cocultures were compared in Figure <a href="#fig:ratio-change">2</a>.</p>
<pre class="r"><code># calculate ratio in cocultures
ratio &lt;- qPCR_data %&gt;% 
  filter(condition %in% c(&quot;less&quot;,&quot;equal&quot;,&quot;more&quot;)) %&gt;% 
  group_by(sample, condition, species, time) %&gt;% 
  mutate(rep = row_number()) %&gt;% 
  ungroup() %&gt;% 
  pivot_wider(names_from = species, values_from = quantity) %&gt;%
  mutate(ratio = EC/PP) %&gt;%
  filter(!is.na(ratio))</code></pre>
<p>Figure <a href="#fig:ratio-change">2</a> shows the deterministic assembly of <em>E. coli</em> and <em>P. putida</em> cocultures. (A) the real-time ratio of EC: PP after 0, 0.5, 1, 2, 4, 8 and 24 h cultivation. (B) Analysis of the variances of EC/PP ratios (24h) between cocultures.</p>

<pre class="r"><code>ratio.sum &lt;- ratio %&gt;% 
  group_by(sample) %&gt;%
  mutate(y=mean(ratio,na.rm=TRUE),std=sd(ratio,na.rm=TRUE)) %&gt;% 
  dplyr::select(sample,condition, time, y, std) %&gt;%
  unique()
ratio.sum$condition &lt;- factor(ratio.sum$condition,
                        levels = c(&quot;less&quot;,&quot;equal&quot;,&quot;more&quot;),
                        labels = c(&quot;1:1000&quot;,&quot;1:1&quot;,&quot;1000:1&quot;)) 
plot_ratio &lt;- ggplot(ratio.sum, aes(time,y,shape=condition,color=condition)) + 
  geom_rect(aes(xmin=23,xmax=25,ymin=0.02,ymax=0.3),
            fill=&quot;lightyellow&quot;,color=&quot;black&quot;,alpha=0.1) +
  geom_line(size=1,show.legend = F) +
  geom_point(size=2,show.legend = F) +
  geom_errorbar(aes(ymin=y-std,ymax=y+std),show.legend = F) +
  geom_text(aes(x=9,label=condition),hjust=0,vjust=c(0,0,1),
            data = filter(ratio.sum,time==8),
            show.legend = F) +
  scale_y_log10(labels=formatC,breaks=10^(-3:3),
                expand = expansion(0.1)) +
  geom_bracket(xmin = 0, xmax = 2,y.position=3.5,label = &quot;ns&quot;,inherit.aes = FALSE) +
  geom_bracket(xmin = 0, xmax = 4,y.position=1,label = &quot;ns&quot;,inherit.aes = FALSE) +
  geom_bracket(xmin = 0, xmax = 4,y.position=0.001,label = &quot;ns&quot;,inherit.aes = FALSE) +
  labs(x=&quot;Time (h)&quot;, y=&quot;Ratio (EC/PP)&quot;) +
  theme(legend.position = c(0.8,0.75))

ratio_24h &lt;- ratio %&gt;% filter(time==24)
plot_ratio_stats &lt;- ggplot(ratio_24h,aes(condition,ratio,color=condition)) +
  geom_boxplot(fill=&quot;lightyellow&quot;) + 
  geom_jitter() + 
  stat_compare_means(
    comparisons = list(c(&quot;less&quot;,&quot;equal&quot;),c(&quot;less&quot;,&quot;more&quot;),c(&quot;equal&quot;,&quot;more&quot;)),
    label=&quot;p.format&quot;) +
   scale_x_discrete(breaks=c(&quot;less&quot;,&quot;equal&quot;,&quot;more&quot;),
                    labels=c(&quot;1:1000&quot;,&quot;1:1&quot;,&quot;1000:1&quot;))+
  xlab(&quot;Condition&quot;) + ylab(&quot;Ratio (EC/PP)&quot;) +
  scale_y_continuous(expand = expansion(0.2)) +
  theme(legend.position = &quot;none&quot;,
        panel.background = element_rect(fill=&quot;lightyellow&quot;))

plot_grid(plot_ratio,plot_ratio_stats,labels = &quot;auto&quot;, ncol = 3, rel_widths = c(2,1.2,2))</code></pre>
<div class="figure" style="text-align: center"><span id="fig:ratio-change"></span>
<img src="gene-expression_files/figure-html/ratio-change-1.png" alt="Deterministic assembly of E. coli and P. putida cocultures. (A) the real-time ratio of EC: PP after 0, 0.5, 1, 2, 4, 8 and 24 h cultivation. (B) Analysis of the variances of EC/PP ratios (24h) between cocultures." width="75%" />
<p class="caption">
Figure 2: Deterministic assembly of <em>E. coli</em> and <em>P. putida</em> cocultures. (A) the real-time ratio of EC: PP after 0, 0.5, 1, 2, 4, 8 and 24 h cultivation. (B) Analysis of the variances of EC/PP ratios (24h) between cocultures.
</p>
</div>
<pre class="r"><code>export::graph2ppt(append  = TRUE)</code></pre>
<p>Notably, the ratio differences were non-significant by time in the logarithmic phase, but were all significant in the stationary phase for every coculture (Figure <a href="#fig:comparison-by-time">3</a>).</p>
<p>For each coculture, significance of variances between each sample were as follows:</p>
<ul>
<li>the 1:1000 coculture</li>
</ul>
<pre class="r"><code>### stats
ratio1 &lt;- ratio %&gt;% filter(condition==&quot;less&quot;)
(p1 &lt;- pairwise.wilcox.test(ratio1$ratio,ratio1$time,p.adjust.method = &quot;BH&quot;))</code></pre>
<pre><code>## 
##  Pairwise comparisons using Wilcoxon rank sum exact test 
## 
## data:  ratio1$ratio and ratio1$time 
## 
##     0      0.5    1      2      4      8     
## 0.5 0.5359 -      -      -      -      -     
## 1   0.9372 0.4596 -      -      -      -     
## 2   0.0785 0.4062 0.1049 -      -      -     
## 4   0.1848 0.4596 0.1848 0.7341 -      -     
## 8   0.0045 0.0045 0.0045 0.0045 0.0045 -     
## 24  0.0045 0.0045 0.0045 0.0045 0.0045 0.1049
## 
## P value adjustment method: BH</code></pre>
<ul>
<li>the 1:1 coculture</li>
</ul>
<pre class="r"><code>ratio2 &lt;- ratio %&gt;% filter(condition==&quot;equal&quot;)
(p2 &lt;- pairwise.wilcox.test(ratio2$ratio,ratio2$time,p.adjust.method = &quot;BH&quot;))</code></pre>
<pre><code>## 
##  Pairwise comparisons using Wilcoxon rank sum exact test 
## 
## data:  ratio2$ratio and ratio2$time 
## 
##     0      0.5    1      2      4      8     
## 0.5 0.7727 -      -      -      -      -     
## 1   0.2902 0.2197 -      -      -      -     
## 2   0.7273 0.7922 0.3364 -      -      -     
## 4   0.6364 0.7922 0.3364 0.7727 -      -     
## 8   0.0051 0.0083 0.0051 0.0051 0.0051 -     
## 24  0.0051 0.0083 0.0051 0.0051 0.0051 0.0051
## 
## P value adjustment method: BH</code></pre>
<ul>
<li>and the 1000:1 coculture</li>
</ul>
<pre class="r"><code>ratio3 &lt;- ratio %&gt;% filter(condition==&quot;more&quot;)
(p3 &lt;- pairwise.wilcox.test(ratio3$ratio,ratio3$time,p.adjust.method = &quot;BH&quot;))</code></pre>
<pre><code>## 
##  Pairwise comparisons using Wilcoxon rank sum exact test 
## 
## data:  ratio3$ratio and ratio3$time 
## 
##     0      0.5    1      2      4      8     
## 0.5 0.2968 -      -      -      -      -     
## 1   0.6991 0.2358 -      -      -      -     
## 2   0.3636 0.1152 0.4500 -      -      -     
## 4   0.0070 0.0070 0.0130 0.3636 -      -     
## 8   0.0051 0.0051 0.0051 0.0070 0.0051 -     
## 24  0.0051 0.0051 0.0051 0.0070 0.0051 0.0051
## 
## P value adjustment method: BH</code></pre>
<p>Figure <a href="#fig:comparison-by-time">3</a> shows P-values of pairwise comparison of real-time EC/PP ratios in the “1:1000” (A), “1:1” (B) and “1000:1” (C) cocultures. Circles showed the adjusted p-value for the comparison of the EC/PP ratios between two samples, which were indicated on the top and left (large circle mean large p-value). On this plot, a cross mark was given if the p-value is non-significant (P &gt; 0.05).</p>
<pre class="r"><code>par(mfrow=c(1,3))
pvalues &lt;- list(p1,p2,p3)
line = 0
cex = 1.2
side = 3
adj=0.15
plots &lt;- lapply(seq_along(pvalues), function(i){
  x &lt;- pvalues[[i]]
  corrplot(x$p.value, 
           type = &quot;lower&quot;,
           col = &quot;grey&quot;,
           cl.pos = &quot;n&quot;,
           is.corr = FALSE, 
           method = &quot;circle&quot;,
           p.mat = x$p.value,
           sig.level = 0.05)
  mtext(LETTERS[[i]], side = side, line = line, cex = cex, adj = adj)

})</code></pre>
<div class="figure" style="text-align: center"><span id="fig:comparison-by-time"></span>
<img src="gene-expression_files/figure-html/comparison-by-time-1.png" alt="P-values of pairwise comparison of real-time EC/PP ratios in the “1:1000” (A), “1:1” (B) and “1000:1” (C) cocultures." width="75%" />
<p class="caption">
Figure 3: P-values of pairwise comparison of real-time EC/PP ratios in the “1:1000” (A), “1:1” (B) and “1000:1” (C) cocultures.
</p>
</div>
</div>
</div>
<div id="gene-expression-analysis" class="section level2">
<h2>Gene expression analysis</h2>
<p>To reveal the mechanism of community assembly in cocultures, we investigated the transcriptomic changes in cocultures using RNA-seq analysis.</p>
<p>Totally 60 samples were sequences. After sequencing quality control, each sample has 2.6 – 3.9 M paired reads, and 3.9 – 5.9 G base pairs, having an overall coverage of 300 X at least. After filtration, high-quality reads were aligned against the P. putida (<a href="https://www.ncbi.nlm.nih.gov/genome/?term=pseudomonas+putida+kt2440" class="uri">https://www.ncbi.nlm.nih.gov/genome/?term=pseudomonas+putida+kt2440</a>) and E.coli reference genome (<a href="https://www.ncbi.nlm.nih.gov/genome/?term=Escherichia+coli+K-12" class="uri">https://www.ncbi.nlm.nih.gov/genome/?term=Escherichia+coli+K-12</a>) using hisat2 and gene expression changes were quantified using DESeq2 software <span class="citation">(Herberg et al. 2016; Love, Huber, and Anders 2014)</span>. While aligning to reference genomes, the overall aligned rates are ranging from 97.23% to 98.43%.</p>
<pre class="r"><code>tableS1 &lt;- read.xlsx(&quot;./data/tableS1.xlsx&quot;)
summary(tableS1)</code></pre>
<pre><code>##   Condition              Time       Sample          Raw_Read_Number   
##  Length:60          Min.   : 0   Length:60          Min.   :26331150  
##  Class :character   1st Qu.: 3   Class :character   1st Qu.:30196483  
##  Mode  :character   Median : 6   Mode  :character   Median :32429121  
##                     Mean   : 9                      Mean   :32696008  
##                     3rd Qu.:12                      3rd Qu.:34950975  
##                     Max.   :24                      Max.   :39576900  
##    Raw_Bases           Raw_N_rate       Raw_GC_content   Raw_Q20_rate  
##  Min.   :3.976e+09   Min.   :0.000965   Min.   :51.10   Min.   :95.75  
##  1st Qu.:4.560e+09   1st Qu.:0.002455   1st Qu.:51.74   1st Qu.:96.39  
##  Median :4.897e+09   Median :0.004755   Median :55.62   Median :96.56  
##  Mean   :4.937e+09   Mean   :0.005884   Mean   :55.54   Mean   :96.63  
##  3rd Qu.:5.278e+09   3rd Qu.:0.005086   3rd Qu.:59.09   3rd Qu.:96.95  
##  Max.   :5.976e+09   Max.   :0.016776   Max.   :60.07   Max.   :97.30  
##   Raw_Q30_rate   Trimmed_Read_Number Trimmed_Bases        Useful_read%  
##  Min.   :89.66   Min.   :26217232    Min.   :3.943e+09   Min.   :99.17  
##  1st Qu.:90.98   1st Qu.:29991340    1st Qu.:4.511e+09   1st Qu.:99.38  
##  Median :91.43   Median :32249039    Median :4.859e+09   Median :99.46  
##  Mean   :91.59   Mean   :32514927    Mean   :4.896e+09   Mean   :99.44  
##  3rd Qu.:92.41   3rd Qu.:34741486    3rd Qu.:5.234e+09   3rd Qu.:99.50  
##  Max.   :93.06   Max.   :39286830    Max.   :5.921e+09   Max.   :99.64  
##  Useful_bases%    align_rate%   
##  Min.   :98.40   Min.   :97.23  
##  1st Qu.:99.12   1st Qu.:97.98  
##  Median :99.21   Median :98.10  
##  Mean   :99.15   Mean   :98.03  
##  3rd Qu.:99.27   3rd Qu.:98.22  
##  Max.   :99.34   Max.   :98.43</code></pre>
<p>Figure <a href="#fig:reads-count">4</a> shows the reads count of <em>E. coli</em> and <em>P. putida</em> in each RNA-seq library. Samples were taken with each condition at indicated times (at 0, 4, 8 and 24h). (A-D) <em>P. putida</em> monoculture samples; (E-H) <em>E. coli</em> and <em>P. putida</em> “1:1000” coculture; (I-L) the “1:1” coculture; (M-P) the “1000:1” coculture; and (Q-T) the <em>E. coli</em> monoculture samples. Each sample has three replicates. Only aligned reads to either <em>E. coli</em> or <em>P. putida</em> genome were used in this calculation. Plots showed the proportion of reads which have aligned to corresponding genome by different colors, as indicated on the top.</p>
<pre class="r"><code>ht_counts &lt;- readRDS(file = &quot;./data/ht_counts.rds&quot;)
ht_counts$group &lt;- factor(ht_counts$group,
                          levels = c(&quot;none_0h&quot;,&quot;none_4h&quot;,&quot;none_8h&quot;,&quot;none_24h&quot;,&quot;less_0h&quot;,&quot;less_4h&quot;,&quot;less_8h&quot;,&quot;less_24h&quot;,&quot;equal_0h&quot;,&quot;equal_4h&quot;,&quot;equal_8h&quot;,&quot;equal_24h&quot;,&quot;more_0h&quot;,&quot;more_4h&quot;,&quot;more_8h&quot;,&quot;more_24h&quot;,&quot;all_0h&quot;,&quot;all_4h&quot;,&quot;all_8h&quot;,&quot;all_24h&quot;),
                          labels = c(&quot;P. putida_0h&quot;,&quot;P. putida_4h&quot;,&quot;P. putida_8h&quot;,&quot;P. putida_24h&quot;,&quot;1:1000_0h&quot;,&quot;1:1000_4h&quot;,&quot;1:1000_8h&quot;,&quot;1:1000_24h&quot;,&quot;1:1_0h&quot;,&quot;1:1_4h&quot;,&quot;1:1_8h&quot;,&quot;1:1_24h&quot;,&quot;1000:1_0h&quot;,&quot;1000:1_4h&quot;,&quot;1000:1_8h&quot;,&quot;1000:1_24h&quot;,&quot;E. coli_0h&quot;,&quot;E. coli_4h&quot;,&quot;E. coli_8h&quot;,&quot;E. coli_24h&quot;))
ht_counts_total &lt;- ht_counts %&gt;% group_by(sample_id, group, organism) %&gt;%
  summarise(sum_of_reads=sum(count)) %&gt;%
  group_by(sample_id) %&gt;% 
  mutate(proportion=sum_of_reads/sum(sum_of_reads))
samples &lt;- levels(ht_counts_total$group)
plots &lt;- lapply(1:length(samples),function(i){
  sample_group &lt;- samples[[i]]
  df &lt;- filter(ht_counts_total,group==sample_group)
  ggplot(df,aes(x=sample_id, y=proportion,fill=organism)) + 
    geom_bar(stat = &quot;identity&quot;,position = &quot;stack&quot;) + 
    scale_y_continuous(labels = function(l)paste(format(l*100,digits = 2),&quot;%&quot;,sep=&quot;&quot;)) +
    scale_x_discrete(labels=c(&quot;Rep1&quot;,&quot;Rep2&quot;,&quot;Rep3&quot;)) +
    scale_fill_discrete(name=&quot;Organism: &quot;,labels=c(&quot;EC&quot;=&quot;E. coli&quot;,&quot;PP&quot;=&quot;P. putida&quot;)) +
    labs(title = sample_group) +
    theme(legend.text = element_text(face = &quot;italic&quot;),
          legend.position = &quot;none&quot;,
          axis.title = element_blank())
})
legend &lt;- get_legend(plots[[1]] + theme(legend.position = &quot;top&quot;,legend.direction = &quot;horizontal&quot;))
plot_grid(legend, plot_grid(plotlist = plots,labels = &quot;auto&quot;,ncol=4),rel_heights = c(1,15),ncol=1)</code></pre>
<div class="figure" style="text-align: center"><span id="fig:reads-count"></span>
<img src="gene-expression_files/figure-html/reads-count-1.png" alt="Reads count of *E. coli* and *P. putida* in each RNA-seq library" width="75%" />
<p class="caption">
Figure 4: Reads count of <em>E. coli</em> and <em>P. putida</em> in each RNA-seq library
</p>
</div>
<div id="identify-gene-expression-changes" class="section level3">
<h3>Identify gene expression changes</h3>
<p>This step is a time consuming step. Use the precalculated DEG if possible.</p>
<pre class="r"><code># load precalculated DEG results.
dds.EC &lt;- readRDS(&quot;data/dds.EC.2.rds&quot;)
dds.PP &lt;- readRDS(&quot;data/dds.PP.2.rds&quot;)
DEG_results.EC &lt;- readRDS(&quot;data/DEG_results.EC.rds&quot;)
DEG_results.PP &lt;- readRDS(&quot;data/DEG_results.PP.rds&quot;)</code></pre>
</div>
<div id="rna-seq-clustering" class="section level3">
<h3>RNA-seq clustering</h3>
<p>First of all, we compared the gene expression profiles between monoculture and cocultures at genome level. Figure <a href="#fig:RNA-seq-PCA">5</a> shows the Principle coordinate analysis (PCA) of the gene expression profiles of <em>E. coli</em> (A) and <em>P. putida</em> (B) in different samples.</p>
<pre class="r"><code>list_of_vsd &lt;- lapply(list(dds.EC,dds.PP),function(dds){
  vst(dds,blind = F)
})
list_of_vsd[[1]]$ratio0 &lt;- factor(list_of_vsd[[1]]$ratio0,
                                  levels = c(&quot;none&quot;,&quot;less&quot;,&quot;equal&quot;,&quot;more&quot;,&quot;all&quot;),
                        labels = c(&quot;P. putida&quot;,&quot;1:1000&quot;,&quot;1:1&quot;,&quot;1000:1&quot;,&quot;E. coli&quot;))
list_of_vsd[[2]]$ratio0 &lt;- factor(list_of_vsd[[2]]$ratio0,
                                  levels = c(&quot;none&quot;,&quot;less&quot;,&quot;equal&quot;,&quot;more&quot;,&quot;all&quot;),
                        labels = c(&quot;P. putida&quot;,&quot;1:1000&quot;,&quot;1:1&quot;,&quot;1000:1&quot;,&quot;E. coli&quot;))</code></pre>
<p>The results showed that the gene expression profiles were different between monoculture and cocultures in the beginning, and became more consistent in the later (Fig. <a href="#fig:RNA-seq-PCA">5</a>A). Differences in gene expression appeared immediately after mixing of <em>E. coli</em> and <em>P. putida</em>, which has been described by 0h data (due to the time required for manual operation, this is of course not the real 0h). The expression profiles (0h) of <em>E. coli</em> monoculture, “1:1” and “1000:1” cocultures are close, but the “1:1000” coculture is distinct from the others. At 4h and 8h, we can still visually separate the “less” samples and the others on the plots. However, the expression profiles of <em>E. coli</em> became closer by time. After 24h cultivation, there was no obvious boundary between <em>E. coli</em> monoculture and three cocultures. Likewise, the expression profiles of <em>P. putida</em> has a similar pattern, except that the distance between “1000:1” coculture and other three samples was more obvious (Fig. <a href="#fig:RNA-seq-PCA">5</a>B).</p>
<pre class="r"><code>myPlotPCA &lt;- function(object,
                      intgroup = c(&quot;time&quot;,&quot;ratio0&quot;), 
                      show.label = FALSE,
                      return_data = FALSE) {
  require(dplyr,quietly = T)
  require(DESeq2,quietly = T)
  require(vegan,quietly = T)
  pca &lt;- rda(t(assay(object))) 
  
  percent_var &lt;- pca$CA$eig/pca$tot.chi  
  
  if (!all(intgroup %in% names(colData(object)))) {
    stop(&quot;the argument &#39;intgroup&#39; should specify columns of colData(dds)&quot;)
  }
  intgroup_df &lt;- as.data.frame(colData(object)[, intgroup, 
                                               drop = FALSE]) %&gt;%
    tibble::rownames_to_column(var = &quot;sample_id&quot;)
  
  df &lt;- scores(pca)$sites %&gt;% 
    as.data.frame() %&gt;%
    tibble::rownames_to_column(var=&quot;sample_id&quot;) %&gt;%
    left_join(intgroup_df,by=&quot;sample_id&quot;) %&gt;% 
    mutate(time=factor(time,levels = sort(unique(as.numeric(time)))))
  
  if (return_data){
    attr(df, &quot;percentvar&quot;) &lt;- percent_var
    return(df)
  } 
  
  mapping &lt;- aes(PC1, PC2, color=ratio0, label=sample_id)
  
  p &lt;- ggplot(df,mapping) +
    geom_point(size=2)  +
    xlab(paste0(&quot;PC1: &quot;, round(percent_var[1] * 100), &quot;% variance&quot;)) + 
    ylab(paste0(&quot;PC2: &quot;, round(percent_var[2] * 100), &quot;% variance&quot;))
  
  if (show.label) {
    return(p + geom_text_repel(show.legend = F))
  } else {
    return(p)
  }
  
}</code></pre>
<pre class="r"><code>list_of_PCA_plot &lt;- lapply(list_of_vsd, function(vsd) {
  myPlotPCA(vsd) + 
    facet_wrap(~time,ncol=4) +
    directlabels::geom_dl(aes(label=ratio0),method = &quot;smart.grid&quot;,size=2) +  #文本代替标签 位置标注的不好,改size没用
    scale_color_manual(limits=c(&quot;E. coli&quot;,&quot;1:1000&quot;,&quot;1:1&quot;,&quot;1000:1&quot;,&quot;P. putida&quot;),
                       values = brewer.pal(5,&quot;Dark2&quot;)) +
    theme(legend.position = &quot;none&quot;)
  })

plot_grid(plotlist = list_of_PCA_plot,labels = &quot;auto&quot;,ncol=1)</code></pre>
<div class="figure" style="text-align: center"><span id="fig:RNA-seq-PCA"></span>
<img src="gene-expression_files/figure-html/RNA-seq-PCA-1.png" alt="Principle coordinate analysis (PCA) of the gene expression profiles of *E. coli* (A) and *P. putida* (B) in different samples." width="75%" />
<p class="caption">
Figure 5: Principle coordinate analysis (PCA) of the gene expression profiles of <em>E. coli</em> (A) and <em>P. putida</em> (B) in different samples.
</p>
</div>
</div>
<div id="beta-dispersion" class="section level3">
<h3>Beta-dispersion</h3>
<p>According to one of the reviewers’ suggestions, we <strong>use the beta-dispersion</strong> to compare the significance of gene expression change within different time points.</p>
<pre class="r"><code>models = lapply(list_of_vsd, function(object){
  dist = vegdist(t(assay(object)), method = &quot;euclidean&quot;)
  group_data = colData(object) %&gt;% 
    as_tibble() %&gt;%
    dplyr::mutate(group=paste0(ratio0,&quot; (&quot;,time,&quot;h)&quot;)) 
  mod = with(group_data, betadisper(dist, group = group))
  return(mod)
})</code></pre>
<p>We define a function <code>ggplot.betadisper()</code> to plot <em>betadisper</em> object with ggplot2 methods.</p>
<pre class="r"><code># group meta data
group_meta = tableS1 %&gt;%
  dplyr::rename_all(tolower) %&gt;%
  dplyr::select(sample, condition, time) %&gt;%
  dplyr::mutate(condition = factor(condition,
                               levels = c(&quot;none&quot;,&quot;less&quot;,&quot;equal&quot;,&quot;more&quot;,&quot;all&quot;),
                               labels = c(&quot;P. putida&quot;,&quot;1:1000&quot;,&quot;1:1&quot;,&quot;1000:1&quot;,&quot;E. coli&quot;))) %&gt;%
  dplyr::mutate(group =  paste0(condition,&quot; (&quot;,time,&quot;h)&quot;))

ggplot.betadisper = function(x){
  
  percent_var = x$eig/sum(x$eig)
  xylab = paste0(&quot;PCoA &quot;, 1:2, &quot;: &quot;, round(percent_var[1:2] * 100), &quot;% variance&quot;) 
  
  sites = scores(x, display = &quot;sites&quot;, choices = c(1,2)) %&gt;% 
    as.data.frame() %&gt;% 
    tibble::rownames_to_column(var=&quot;sample&quot;)
  sites$group = x$group
  sites = left_join(sites, group_meta)
  
  centroids = scores(x, display = &quot;centroids&quot;, choices = 1:2) %&gt;%
    as.data.frame() %&gt;%
    tibble::rownames_to_column(var=&quot;group&quot;)
  centroids = left_join(centroids, group_meta %&gt;% select(-sample) %&gt;% unique())
  
  ggplot() +
    aes_string(x=&quot;PCoA1&quot;,y=&quot;PCoA2&quot;) +
    geom_point(aes(color = condition), shape = 21, alpha = 1/2,data = sites) +
    geom_polygon(aes(fill=condition, group=group), data=sites, alpha=1/2) +
    geom_point(aes(color=condition, fill=condition), shape=21, size = 3, data=centroids ) +
    ggrepel::geom_label_repel(aes(label = group, color = condition), 
                              data=centroids, 
                              alpha=1/2,
                              label.size = NA,
                              segment.color = &quot;grey&quot;,
                              fontface = &quot;bold.italic&quot;) + 
    scale_color_manual(limits=c(&quot;E. coli&quot;,&quot;1:1000&quot;,&quot;1:1&quot;,&quot;1000:1&quot;,&quot;P. putida&quot;),
                       values = brewer.pal(5,&quot;Dark2&quot;)) +
    scale_fill_manual(limits=c(&quot;E. coli&quot;,&quot;1:1000&quot;,&quot;1:1&quot;,&quot;1000:1&quot;,&quot;P. putida&quot;),
                       values = brewer.pal(5,&quot;Dark2&quot;)) +
    labs(x = xylab[[1]], 
         y = xylab[[2]]) +
    theme(legend.position = &quot;none&quot;)
  

}</code></pre>

<pre class="r"><code>p1 = ggplot.betadisper(models[[1]])
p1</code></pre>
<div class="figure" style="text-align: center"><span id="fig:betadisper-ec"></span>
<img src="gene-expression_files/figure-html/betadisper-ec-1.png" alt="0.5" width="75%" />
<p class="caption">
Figure 6: 0.5
</p>
</div>
<pre class="r"><code>export::graph2ppt(append  = TRUE)</code></pre>

<pre class="r"><code>p2 = ggplot.betadisper(models[[2]])
p2</code></pre>
<div class="figure" style="text-align: center"><span id="fig:betadisper-pp"></span>
<img src="gene-expression_files/figure-html/betadisper-pp-1.png" alt="0.5" width="75%" />
<p class="caption">
Figure 7: 0.5
</p>
</div>
<pre class="r"><code>export::graph2ppt(append  = TRUE)</code></pre>
<div id="comparison-of-dispersion" class="section level4">
<h4>Comparison of dispersion</h4>
<ul>
<li>For the dispersion of <em>E. coli</em></li>
</ul>
<p>The gene expression dispersion of 1:1000 is the highest at 0 and 4 h.</p>
<pre class="r"><code>par(mar = c(7,5,2,1))
boxplot(models[[1]], las = 2, xlab = NA, 
        main = expression( italic(E.~coli )))</code></pre>
<p><img src="gene-expression_files/figure-html/unnamed-chunk-13-1.png" width="75%" style="display: block; margin: auto;" /></p>
<ul>
<li>For the dispersion of <em>P. putida</em></li>
</ul>
<p>The gene expression dispersion of 1000:1 is the highest at 0, 4 and 8 h.</p>
<pre class="r"><code>par(mar = c(7,5,2,1))
boxplot(models[[2]], las = 2, xlab = NA, 
        main = expression(italic(P.~putida)))</code></pre>
<p><img src="gene-expression_files/figure-html/unnamed-chunk-14-1.png" width="75%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="number-of-degs" class="section level3">
<h3>Number of DEGs</h3>
<p>Figure <a href="#fig:number-of-deg">8</a> shows the numbers of differentially expressed genes (DEGs) in three cocultures, in <em>E. coli</em> (A) and <em>P. putida</em> (B). The DEGs were identified by comparison with the corresponding monoculture at the same time. Up- and down-regulation of genes were colored by red and cyan, respectively.</p>
<pre class="r"><code>## count DEG
deg_count &lt;- function(data){
  do.call(&quot;rbind&quot;,lapply(data, function(x) table(x$expression))) %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(var = &quot;name&quot;) %&gt;%
    separate(name, into = c(&quot;ratio&quot;,&quot;time&quot;), sep = &quot;_&quot;, extra = &quot;drop&quot;) %&gt;%
    mutate(time = as.numeric(str_extract(time, &quot;[0-9]+&quot;))) %&gt;%
    pivot_longer(cols = c(&quot;dn&quot;,&quot;up&quot;), names_to = &quot;type&quot;, values_to = &quot;count&quot;) %&gt;%
    mutate(count = ifelse(type == &quot;dn&quot;, -count, count)) %&gt;%
    complete(ratio, time, type, fill = list(count = 0)) %&gt;%
    mutate(ratio = factor(ratio, 
                          levels = c(&quot;less&quot;,&quot;equal&quot;,&quot;more&quot;),
                          labels = c(&quot;1:1000&quot;,&quot;1:1&quot;,&quot;1000:1&quot;)),
           type = factor(type,
                         levels = c(&quot;up&quot;,&quot;dn&quot;),
                         labels = c(&quot;Up&quot;,&quot;Down&quot;)))
}


deg_count_EC &lt;- deg_count(DEG_results.EC)
deg_count_PP &lt;- deg_count(DEG_results.PP)
count &lt;- list(deg_count_EC, deg_count_PP)
library(ggtext)
deg_count_plots &lt;- lapply(seq_along(count), function(i){
  x &lt;- count[[i]]
  ggplot(x, aes(x = time, y = count, color = type)) +
    geom_point() +
    geom_line(size = 1) +
    scale_y_continuous(labels = function(x){abs(x)}) +
    facet_wrap(~ratio) +
    labs(x=&quot;Time(h)&quot;,y=&quot;Number of DEGs&quot;,
         color = &quot;Gene expression:&quot;,
         title= paste0(&quot;DEGs in *&quot;,organsim_fullname[[i]],&quot;*&quot;)) +
    theme(legend.position = c(0.618,1),
        legend.justification = c(0.5,-0.65),
        legend.direction = &quot;horizontal&quot;,
        plot.title = element_markdown())
}) 


plot_grid(plotlist = deg_count_plots,labels = &quot;auto&quot;,ncol = 1)</code></pre>
<div class="figure" style="text-align: center"><span id="fig:number-of-deg"></span>
<img src="gene-expression_files/figure-html/number-of-deg-1.png" alt="numbers of differentially expressed genes (DEGs) in three cocultures, in *E. coli* (A) and *P. putida* (B)." width="75%" />
<p class="caption">
Figure 8: numbers of differentially expressed genes (DEGs) in three cocultures, in <em>E. coli</em> (A) and <em>P. putida</em> (B).
</p>
</div>
</div>
<div id="specific-degs-in-e.-coli-and-p.-putida" class="section level3">
<h3>Specific DEGs in <em>E. coli</em> and <em>P. putida</em></h3>
<p>Figure <a href="#fig:deg-venn">9</a> shows the comparision of differentially expressed genes by time in <em>E. coli</em> (A-C) and <em>P. putida</em> (D-F). Although DEGs overlapped for different time, specific genes are the majority in almost every time point. For instance, 57 out of 93 DEGs are specific in 1:1000 coculture in <em>E. coli</em> (Fig. <a href="#fig:deg-venn">9</a>A).</p>
<pre class="r"><code>library(ggVennDiagram)

ratio0 &lt;- c(&quot;1:1000&quot;,&quot;1:1&quot;,&quot;1000:1&quot;)
deg_Venn_plot_EC &lt;- lapply(seq_along(ratio0), function(i){
  gene_list &lt;- lapply(DEG_results.EC[(i*4-3):(i*4)], function(x){x$gene})
  ggVennDiagram(gene_list,label = &quot;count&quot;,
                category.names = c(&quot;0h&quot;,&quot;4h&quot;,&quot;8h&quot;,&quot;24h&quot;)) +
    scale_fill_gradient(low=&quot;white&quot;,high=&quot;red&quot;,limits=c(0,310)) +
    labs(title=paste0(ratio0[[i]],&quot; - *E. coli*&quot;)) +
    theme(legend.position = &quot;none&quot;,
          plot.title = element_markdown(hjust=0.5))
})

deg_Venn_plot_PP &lt;- lapply(seq_along(ratio0), function(i){
  gene_list &lt;- lapply(DEG_results.PP[(i*4-3):(i*4)], function(x){x$gene})
  ggVennDiagram(gene_list,label = &quot;count&quot;,
                category.names = c(&quot;0h&quot;,&quot;4h&quot;,&quot;8h&quot;,&quot;24h&quot;)) +
    scale_fill_gradient(low=&quot;white&quot;,high=&quot;red&quot;,limits=c(0,310)) +
    labs(title=paste0(ratio0[[i]],&quot; - *P. putida*&quot;)) +
    theme(legend.position = &quot;none&quot;,
          plot.title = element_markdown(hjust=0.5))
})

plot_grid(plotlist = c(deg_Venn_plot_EC,deg_Venn_plot_PP),
          labels = &quot;auto&quot;)</code></pre>
<div class="figure" style="text-align: center"><span id="fig:deg-venn"></span>
<img src="gene-expression_files/figure-html/deg-venn-1.png" alt="Comparision of differentially expressed genes by time" width="75%" />
<p class="caption">
Figure 9: Comparision of differentially expressed genes by time
</p>
</div>
</div>
</div>
<div id="enrichment-analysis-of-degs" class="section level2">
<h2>Enrichment analysis of DEGs</h2>
<p>We use <strong>clusterProfiler</strong> to perform KEGG enrichment analysis.</p>
<pre class="r"><code>kegg_path_tree &lt;- function(df, pathway_name, gene_id, sep=&quot;/&quot;){
  pathway &lt;- df %&gt;% 
    separate_rows(all_of(gene_id), sep = sep) %&gt;%
    dplyr::select(all_of(c(pathway_name, gene_id))) %&gt;%
    unique() %&gt;%
    mutate(value = 1) %&gt;%
    pivot_wider(id_cols = pathway_name,
                names_from = gene_id,
                values_from = value,
                values_fill = list(value = 0))
  library(vegan)
  matrix &lt;- as.matrix(column_to_rownames(pathway, pathway_name))
  dist &lt;- vegdist(matrix, method = &quot;jaccard&quot;)
  
  library(ape)
  library(ggtree)
  tree &lt;- bionj(dist)
  p &lt;- ggtree(tree,branch.length = &quot;none&quot;)
  p$data %&gt;%
    filter(isTip) %&gt;%
    arrange(y) %&gt;%
    pull(label)
}

ck_plot &lt;- function(ck, 
                    mapping = aes(time, Description, size = GeneRatio)){
  df &lt;- data.frame(ck) %&gt;%
    mutate(ratio = factor(ratio, levels = c(&quot;less&quot;,&quot;equal&quot;,&quot;more&quot;),
                          labels = c(&quot;1:1000&quot;,&quot;1:1&quot;,&quot;1000:1&quot;)),
           time = factor(time, levels = c(&quot;0h&quot;,&quot;4h&quot;,&quot;8h&quot;,&quot;24h&quot;))) 
  df$Description &lt;- factor(df$Description, 
                           levels = 
                             kegg_path_tree(df,
                                            pathway_name = &quot;Description&quot;,
                                            gene_id = &quot;geneID&quot;))
  df$GeneRatio &lt;- sapply(df$GeneRatio, function(x) eval(parse(text = x)))
  ggplot(df, mapping = mapping) +
    geom_point() +
    facet_grid(~ ratio, scales = &quot;free_y&quot;) +
    labs(y=&quot;KEGG pathway&quot;)
}


grid_panel_autoheight &lt;- function(p){
  # 调整panel的高度
  require(gtable)
  gp &lt;- ggplotGrob(p)
  # gtable_show_layout(gp)
  facet.rows &lt;- gp$layout$t[grepl(&quot;panel&quot;,gp$layout$name)]
  y.var &lt;- sapply(ggplot_build(p)$layout$panel_scales_y,
                  function(l) length(l$range$range))
  gp$heights[facet.rows] &lt;- gp$heights[facet.rows] * y.var
  return(gp)
}

reset_facet_width &lt;- function(p, rel_width = c(1)){
  gp &lt;- ggplotGrob(p)
  facet.cols &lt;- gp$layout$l[grepl(&quot;panel&quot;,gp$layout$name)]
  gp$widths[facet.cols] &lt;- gp$widths[facet.cols] * rel_width
  return(gp)
}</code></pre>
<div id="kegg-enrichment-in-e.-coli" class="section level3">
<h3>KEGG enrichment in <em>E. coli</em></h3>
<pre class="r"><code>deg1 &lt;- do.call(&quot;rbind&quot;, DEG_results.EC) %&gt;% 
  separate(comparison, into = c(&quot;ratio&quot;,&quot;time&quot;), extra = &quot;drop&quot;)

ck1 &lt;- compareCluster(gene ~ ratio + time, 
                      data = deg1, 
                      fun = &quot;enrichKEGG&quot;, 
                      organism = &quot;eco&quot;,
                      use_internal_data = TRUE) 

p1 &lt;- ck_plot(ck1)
p1 &lt;- grid_panel_autoheight(p1)</code></pre>
</div>
<div id="kegg-enrichment-in-p.-putida" class="section level3">
<h3>KEGG enrichment in <em>P. putida</em></h3>
<pre class="r"><code>deg2 &lt;- do.call(&quot;rbind&quot;, DEG_results.PP) %&gt;% 
  separate(comparison, into = c(&quot;ratio&quot;,&quot;time&quot;), extra = &quot;drop&quot;)

ck2 &lt;- compareCluster(gene ~ ratio + time, 
                      data = deg2, 
                      fun = &quot;enrichKEGG&quot;, 
                      organism = &quot;ppu&quot;,
                      use_internal_data = TRUE) 

p2 &lt;- ck_plot(ck2)
p2 &lt;- grid_panel_autoheight(p2)</code></pre>
</div>
<div id="dotplot-of-kegg-enrichment-results" class="section level3">
<h3>Dotplot of KEGG enrichment results</h3>

<pre class="r"><code>plot_grid(p1,p2, rel_heights = c(1,0.3), ncol = 1, labels = &quot;auto&quot;,align = &quot;v&quot;)</code></pre>
<pre><code>## Warning: Graphs cannot be vertically aligned unless the axis parameter is set.
## Placing graphs unaligned.</code></pre>
<div class="figure" style="text-align: center"><span id="fig:kegg-ora"></span>
<img src="gene-expression_files/figure-html/kegg-ora-1.png" alt="A dot plot shows the KEGG enrichment result of E. coli (A) and P. putida (B) in the three coculture as a function of time. " width="75%" />
<p class="caption">
Figure 10: A dot plot shows the KEGG enrichment result of <em>E. coli</em> (A) and <em>P. putida</em> (B) in the three coculture as a function of time.
</p>
</div>
</div>
</div>
<div id="get-set-enrichment-analysis-gsea-of-gene-expression-profile" class="section level2">
<h2>Get set enrichment analysis (GSEA) of gene expression profile</h2>
<p>Enrichment analysis of DEGs is a common approach in analyzing gene expression profiles. This approach will find genes where the difference is large, but it will not detect a situation where the difference is small, but evidenced in coordinated way in a set of related genes. Gene Set Enrichment Analysis (GSEA) <span class="citation">(Subramanian et al. 2005)</span> directly addresses this limitation. All genes can be used in GSEA; GSEA aggregates the per gene statistics across genes within a gene set, therefore making it possible to detect situations where all genes in a predefined set change in a small but coordinated way. Since it is likely that many relevant phenotypic differences are manifested by small but consistent changes in a set of genes.</p>
<pre class="r"><code># load pre-calculated result
gene_expression.EC &lt;- readRDS(&quot;data/gene_expression.EC.rds&quot;)
gene_expression.PP &lt;- readRDS(&quot;data/gene_expression.PP.rds&quot;)</code></pre>
<pre class="r"><code># define a function to extract and prepare formatted data for GSEA
get_genelist &lt;- function(x){
  if (nrow(x) &lt; 1) return(NULL)
  geneList &lt;- x$log2FoldChange
  names(geneList) &lt;- x$gene
  geneList &lt;- sort(geneList,decreasing = T) 
  return(geneList)
}

set.seed(1234)</code></pre>
<div id="e.-coli-gsea-kegg-result" class="section level3">
<h3><em>E. coli</em> GSEA KEGG result</h3>
<pre class="r"><code>gseKEGG_results.EC &lt;- lapply(gene_expression.EC, function(x){
  geneList &lt;- get_genelist(x)
  tryCatch(gseKEGG(geneList, 
                   organism = &quot;eco&quot;,
                   eps = 1e-20,
                   pvalueCutoff = 1, # all results
                   use_internal_data = TRUE),
           error=function(e) NULL)
})
count.EC.pathways = nrow(data.frame(gseKEGG_results.EC[[1]]))</code></pre>
<p>The GSEA analysis of <em>E. coli</em> included 81 pathways.</p>
</div>
<div id="p.-putida-gsea-kegg-result" class="section level3">
<h3><em>P. putida</em> GSEA KEGG result</h3>
<pre class="r"><code>gseKEGG_results.PP &lt;- lapply(gene_expression.PP, function(x){
  geneList &lt;- get_genelist(x)
  tryCatch(gseKEGG(geneList, 
                   organism = &quot;ppu&quot;,
                   eps = 1e-20,
                   pvalueCutoff = 1, # all results
                   use_internal_data = TRUE),
           error=function(e) NULL)
})
count.PP.pathways = nrow(data.frame(gseKEGG_results.PP[[1]]))</code></pre>
<p>The GSEA analysis of <em>P. putida</em> included 82 pathways.</p>
</div>
<div id="dotplot-of-gsea-results" class="section level3">
<h3>Dotplot of GSEA results</h3>
<pre class="r"><code># combine and reform GSEA result to a data frame
gse_result &lt;- function(result){
  name &lt;- names(result)
  l &lt;- lapply(seq_along(result), function(i){
    data.frame(result[[i]]) %&gt;%
      mutate(comparison = name[[i]])
  })
  do.call(&quot;rbind&quot;, l) %&gt;%
    separate(comparison, into = c(&quot;ratio&quot;,&quot;time&quot;), extra = &quot;drop&quot;) %&gt;%
    mutate(ratio = factor(ratio, levels = c(&quot;less&quot;,&quot;equal&quot;,&quot;more&quot;),
                          labels = c(&quot;1:1000&quot;,&quot;1:1&quot;,&quot;1000:1&quot;)),
           time = factor(time, levels = c(&quot;0h&quot;,&quot;4h&quot;,&quot;8h&quot;,&quot;24h&quot;)),
           type = ifelse(p.adjust &gt; 0.05, &quot;unchanged&quot;,
                         ifelse(enrichmentScore &gt;0, &quot;activated&quot;, &quot;suppressed&quot;)),
           enrichScore = abs(enrichmentScore)) 
}

# dotplot GSEA result
gse_dotplot &lt;- function(df){
  ggplot(df, aes(time, Description, size = enrichScore,color=type)) +
    geom_point() +
    facet_grid(~ ratio, scales = &quot;free_y&quot;) +
    labs(y=&quot;KEGG pathway&quot;) +
    scale_size(limits = c(0.2,1.0))
}</code></pre>

<pre class="r"><code>plot_grid(gsea_plot_EC,gsea_plot_PP,align = &quot;v&quot;, ncol = 1,labels = &quot;auto&quot;,rel_heights = c(1.5,1))</code></pre>
<div class="figure" style="text-align: center"><span id="fig:gsea-dotplot"></span>
<img src="gene-expression_files/figure-html/gsea-dotplot-1.png" alt="GSEA result of cocultures in E. coli (A) and P. putida (B)" width="75%" />
<p class="caption">
Figure 11: GSEA result of cocultures in <em>E. coli</em> (A) and <em>P. putida</em> (B)
</p>
</div>
<pre class="r"><code>export::graph2ppt(append  = TRUE)</code></pre>
<p>Figure <a href="#fig:gsea-dotplot">11</a> shows the GSEA analysis result for <em>E. coli</em> (A) and <em>P. putida</em> (B).</p>
</div>
<div id="overlap-of-gsea-pathway-in-e.-coli-and-p.-putida" class="section level3">
<h3>Overlap of GSEA pathway in <em>E. coli</em> and <em>P. putida</em></h3>
<pre class="r"><code># todo
merged_gsea &lt;- rbind(mutate(df1, organism = &quot;eco&quot;),
                     mutate(df2, organism = &quot;ppu&quot;))</code></pre>
<pre class="r"><code>upset_gsea = merged_gsea %&gt;% group_by(organism,type,time) %&gt;% nest() 
upset_gsea$pathway &lt;- sapply(upset_gsea$data, function(x) pull(x, Description) %&gt;% as.character())
l &lt;- upset_gsea$pathway
names(l) = paste(upset_gsea$organism, upset_gsea$type, upset_gsea$time,sep = &quot;-&quot;)
upset(fromList(l), nsets=23,keep.order = T, order.by = &quot;freq&quot;)</code></pre>
<p>Figure <a href="#fig:gsea-overlap-vennplot">12</a> shows the Overlap of GSEA pathway in <em>E. coli</em> and <em>P. putida</em>.</p>

<pre class="r"><code># changed pathways
venn_gsea &lt;- merged_gsea %&gt;%
  filter(type %in% c(&quot;suppressed&quot;,&quot;activated&quot;)) %&gt;% 
  group_by(organism, type) %&gt;% nest()
venn_gsea$pathway &lt;- sapply(venn_gsea$data, function(x) pull(x, Description) %&gt;% as.character())

l &lt;- venn_gsea$pathway
names(l) &lt;- paste(venn_gsea$organism, venn_gsea$type,sep = &quot;-&quot;)
gsea_pathway_venn1 &lt;- ggVennDiagram(l,label = &quot;count&quot;) +
    scale_x_continuous(expand = c(0.1,0.1))+ 
  scale_fill_gradient(low=&quot;white&quot;,high=&quot;red&quot;,limits=c(0,100)) +
  theme(legend.position = &quot;none&quot;)

l2 &lt;- list(eco=c(l[[1]],l[[2]]), ppu=c(l[[3]],l[[4]]))
gsea_pathway_venn2 &lt;- ggVennDiagram(l2, label = &quot;count&quot;)+  
  scale_fill_gradient(low=&quot;white&quot;,high=&quot;red&quot;,limits=c(0,50)) +
  theme(legend.position = &quot;none&quot;)

l3 &lt;- list(activated=c(l[[1]],l[[3]]), suppressed=c(l[[2]],l[[4]]))
gsea_pathway_venn3 &lt;- ggVennDiagram(l3, label = &quot;count&quot;) + 
  scale_fill_gradient(low=&quot;white&quot;,high=&quot;red&quot;,limits=c(0,50)) +
 theme(legend.position = &quot;none&quot;)

plot_grid(plot_grid(gsea_pathway_venn2,gsea_pathway_venn3,ncol = 1,labels = c(&quot;A&quot;,&quot;B&quot;)),gsea_pathway_venn1,labels = c(&quot;&quot;,&quot;C&quot;),ncol = 2,rel_widths = c(.5,1))</code></pre>
<div class="figure" style="text-align: center"><span id="fig:gsea-overlap-vennplot"></span>
<img src="gene-expression_files/figure-html/gsea-overlap-vennplot-1.png" alt="Overlap of GSEA pathway in E. coli and P. putida. Pathways were distinguished by their descriptions, and separated to activated and suppressed ones." width="75%" />
<p class="caption">
Figure 12: Overlap of GSEA pathway in <em>E. coli</em> and <em>P. putida</em>. Pathways were distinguished by their descriptions, and separated to activated and suppressed ones.
</p>
</div>
<pre class="r"><code>export::graph2ppt(append  = TRUE)</code></pre>
<p>View intersect in Venn diagram interactively (Figure <a href="#fig:overlap-plotly">13</a>).</p>
<pre class="r"><code>p &lt;- ggVennDiagram(l, label = NULL, show_intersect = TRUE)
plotly::ggplotly(p)</code></pre>
<div class="figure" style="text-align: center"><span id="fig:overlap-plotly"></span>
<div id="htmlwidget-643512ced968260f6b97" style="width:100%;height:474.624px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-643512ced968260f6b97">{"x":{"data":[{"x":[0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,null,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,null,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,null,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,null,0.0214943938344883,0.0303030303030303,0.0404040404040404,0.0505050505050505,0.0606060606060606,0.0707070707070707,0.0808080808080808,0.0909090909090909,0.101010101010101,0.111111111111111,0.121212121212121,0.131313131313131,0.141414141414141,0.151515151515151,0.161616161616162,0.171717171717172,0.181818181818182,0.191919191919192,0.202020202020202,0.212121212121212,0.222222222222222,0.232323232323232,0.242424242424242,0.252525252525253,0.262626262626263,0.272727272727273,0.282828282828283,0.292929292929293,0.303030303030303,0.313131313131313,0.323232323232323,0.333333333333333,0.343434343434343,0.353535353535354,0.363636363636364,0.373737373737374,0.383838383838384,0.393939393939394,0.404040404040404,0.414141414141414,0.424242424242424,0.434343434343434,0.444444444444444,0.454545454545455,0.464646464646465,0.474747474747475,0.484848484848485,0.494949494949495,0.505050505050505,0.515151515151515,0.525252525252525,0.535353535353535,0.545454545454546,0.555555555555556,0.565656565656566,0.575757575757576,0.585858585858586,0.595959595959596,0.606060606060606,0.616161616161616,0.626262626262626,0.636363636363636,0.646464646464647,0.656565656565656,0.666666666666667,0.676767676767677,0.686868686868687,0.696969696969697,0.707070707070707,0.717171717171717,0.727272727272727,0.737373737373737,0.747474747474748,0.757575757575758,0.767676767676768,0.777777777777778,0.787878787878788,0.797979797979798,0.808080808080808,0.818181818181818,0.828282828282828,0.838383838383838,0.848484848484848,0.858585858585859,0.868686868686869,0.878787878787879,0.888888888888889,0.898989898989899,0.909090909090909,0.919191919191919,0.929292929292929,0.939393939393939,0.94949494949495,0.95959595959596,0.96969696969697,0.978505606165512,null,0.0214943938344883,0.0303030303030303,0.0404040404040404,0.0505050505050505,0.0606060606060606,0.0707070707070707,0.0808080808080808,0.0909090909090909,0.101010101010101,0.111111111111111,0.121212121212121,0.131313131313131,0.141414141414141,0.151515151515151,0.161616161616162,0.171717171717172,0.181818181818182,0.191919191919192,0.202020202020202,0.212121212121212,0.222222222222222,0.232323232323232,0.242424242424242,0.252525252525253,0.262626262626263,0.272727272727273,0.282828282828283,0.292929292929293,0.303030303030303,0.313131313131313,0.323232323232323,0.333333333333333,0.343434343434343,0.353535353535354,0.363636363636364,0.373737373737374,0.383838383838384,0.393939393939394,0.404040404040404,0.414141414141414,0.424242424242424,0.434343434343434,0.444444444444444,0.454545454545455,0.464646464646465,0.474747474747475,0.484848484848485,0.494949494949495,0.505050505050505,0.515151515151515,0.525252525252525,0.535353535353535,0.545454545454546,0.555555555555556,0.565656565656566,0.575757575757576,0.585858585858586,0.595959595959596,0.606060606060606,0.616161616161616,0.626262626262626,0.636363636363636,0.646464646464647,0.656565656565656,0.666666666666667,0.676767676767677,0.686868686868687,0.696969696969697,0.707070707070707,0.717171717171717,0.727272727272727,0.737373737373737,0.747474747474748,0.757575757575758,0.767676767676768,0.777777777777778,0.787878787878788,0.797979797979798,0.808080808080808,0.818181818181818,0.828282828282828,0.838383838383838,0.848484848484848,0.858585858585859,0.868686868686869,0.878787878787879,0.888888888888889,0.898989898989899,0.909090909090909,0.919191919191919,0.929292929292929,0.939393939393939,0.94949494949495,0.95959595959596,0.96969696969697,0.978505606165512,null,0.0214943938344883,0.0303030303030303,0.0404040404040404,0.0505050505050505,0.0606060606060606,0.0707070707070707,0.0808080808080808,0.0909090909090909,0.101010101010101,0.111111111111111,0.121212121212121,0.131313131313131,0.141414141414141,0.151515151515151,0.161616161616162,0.171717171717172,0.181818181818182,0.191919191919192,0.202020202020202,0.212121212121212,0.222222222222222,0.232323232323232,0.242424242424242,0.252525252525253,0.262626262626263,0.272727272727273,0.282828282828283,0.292929292929293,0.303030303030303,0.313131313131313,0.323232323232323,0.333333333333333,0.343434343434343,0.353535353535354,0.363636363636364,0.373737373737374,0.383838383838384,0.393939393939394,0.404040404040404,0.414141414141414,0.424242424242424,0.434343434343434,0.444444444444444,0.454545454545455,0.464646464646465,0.474747474747475,0.484848484848485,0.494949494949495,0.505050505050505,0.515151515151515,0.525252525252525,0.535353535353535,0.545454545454546,0.555555555555556,0.565656565656566,0.575757575757576,0.585858585858586,0.595959595959596,0.606060606060606,0.616161616161616,0.626262626262626,0.636363636363636,0.646464646464647,0.656565656565656,0.666666666666667,0.676767676767677,0.686868686868687,0.696969696969697,0.707070707070707,0.717171717171717,0.727272727272727,0.737373737373737,0.747474747474748,0.757575757575758,0.767676767676768,0.777777777777778,0.787878787878788,0.797979797979798,0.808080808080808,0.818181818181818,0.828282828282828,0.838383838383838,0.848484848484848,0.858585858585859,0.868686868686869,0.878787878787879,0.888888888888889,0.898989898989899,0.909090909090909,0.919191919191919,0.929292929292929,0.939393939393939,0.94949494949495,0.95959595959596,0.96969696969697,0.978505606165512,null,0.0214943938344883,0.0303030303030303,0.0404040404040404,0.0505050505050505,0.0606060606060606,0.0707070707070707,0.0808080808080808,0.0909090909090909,0.101010101010101,0.111111111111111,0.121212121212121,0.131313131313131,0.141414141414141,0.151515151515151,0.161616161616162,0.171717171717172,0.181818181818182,0.191919191919192,0.202020202020202,0.212121212121212,0.222222222222222,0.232323232323232,0.242424242424242,0.252525252525253,0.262626262626263,0.272727272727273,0.282828282828283,0.292929292929293,0.303030303030303,0.313131313131313,0.323232323232323,0.333333333333333,0.343434343434343,0.353535353535354,0.363636363636364,0.373737373737374,0.383838383838384,0.393939393939394,0.404040404040404,0.414141414141414,0.424242424242424,0.434343434343434,0.444444444444444,0.454545454545455,0.464646464646465,0.474747474747475,0.484848484848485,0.494949494949495,0.505050505050505,0.515151515151515,0.525252525252525,0.535353535353535,0.545454545454546,0.555555555555556,0.565656565656566,0.575757575757576,0.585858585858586,0.595959595959596,0.606060606060606,0.616161616161616,0.626262626262626,0.636363636363636,0.646464646464647,0.656565656565656,0.666666666666667,0.676767676767677,0.686868686868687,0.696969696969697,0.707070707070707,0.717171717171717,0.727272727272727,0.737373737373737,0.747474747474748,0.757575757575758,0.767676767676768,0.777777777777778,0.787878787878788,0.797979797979798,0.808080808080808,0.818181818181818,0.828282828282828,0.838383838383838,0.848484848484848,0.858585858585859,0.868686868686869,0.878787878787879,0.888888888888889,0.898989898989899,0.909090909090909,0.919191919191919,0.929292929292929,0.939393939393939,0.94949494949495,0.95959595959596,0.96969696969697,0.978505606165512,null,0.0214943938344883,0.0303030303030303,0.0404040404040404,0.0505050505050505,0.0606060606060606,0.0707070707070707,0.0808080808080808,0.0909090909090909,0.101010101010101,0.111111111111111,0.121212121212121,0.131313131313131,0.141414141414141,0.151515151515151,0.161616161616162,0.171717171717172,0.181818181818182,0.191919191919192,0.202020202020202,0.212121212121212,0.222222222222222,0.232323232323232,0.242424242424242,0.252525252525253,0.262626262626263,0.272727272727273,0.282828282828283,0.292929292929293,0.303030303030303,0.313131313131313,0.323232323232323,0.333333333333333,0.343434343434343,0.353535353535354,0.363636363636364,0.373737373737374,0.383838383838384,0.393939393939394,0.404040404040404,0.414141414141414,0.424242424242424,0.434343434343434,0.444444444444444,0.454545454545455,0.464646464646465,0.474747474747475,0.484848484848485,0.494949494949495,0.505050505050505,0.515151515151515,0.525252525252525,0.535353535353535,0.545454545454546,0.555555555555556,0.565656565656566,0.575757575757576,0.585858585858586,0.595959595959596,0.606060606060606,0.616161616161616,0.626262626262626,0.636363636363636,0.646464646464647,0.656565656565656,0.666666666666667,0.676767676767677,0.686868686868687,0.696969696969697,0.707070707070707,0.717171717171717,0.727272727272727,0.737373737373737,0.747474747474748,0.757575757575758,0.767676767676768,0.777777777777778,0.787878787878788,0.797979797979798,0.808080808080808,0.818181818181818,0.828282828282828,0.838383838383838,0.848484848484848,0.858585858585859,0.868686868686869,0.878787878787879,0.888888888888889,0.898989898989899,0.909090909090909,0.919191919191919,0.929292929292929,0.939393939393939,0.94949494949495,0.95959595959596,0.96969696969697,0.978505606165512,null,0.0214943938344883,0.0303030303030303,0.0404040404040404,0.0505050505050505,0.0606060606060606,0.0707070707070707,0.0808080808080808,0.0909090909090909,0.101010101010101,0.111111111111111,0.121212121212121,0.131313131313131,0.141414141414141,0.151515151515151,0.161616161616162,0.171717171717172,0.181818181818182,0.191919191919192,0.202020202020202,0.212121212121212,0.222222222222222,0.232323232323232,0.242424242424242,0.252525252525253,0.262626262626263,0.272727272727273,0.282828282828283,0.292929292929293,0.303030303030303,0.313131313131313,0.323232323232323,0.333333333333333,0.343434343434343,0.353535353535354,0.363636363636364,0.373737373737374,0.383838383838384,0.393939393939394,0.404040404040404,0.414141414141414,0.424242424242424,0.434343434343434,0.444444444444444,0.454545454545455,0.464646464646465,0.474747474747475,0.484848484848485,0.494949494949495,0.505050505050505,0.515151515151515,0.525252525252525,0.535353535353535,0.545454545454546,0.555555555555556,0.565656565656566,0.575757575757576,0.585858585858586,0.595959595959596,0.606060606060606,0.616161616161616,0.626262626262626,0.636363636363636,0.646464646464647,0.656565656565656,0.666666666666667,0.676767676767677,0.686868686868687,0.696969696969697,0.707070707070707,0.717171717171717,0.727272727272727,0.737373737373737,0.747474747474748,0.757575757575758,0.767676767676768,0.777777777777778,0.787878787878788,0.797979797979798,0.808080808080808,0.818181818181818,0.828282828282828,0.838383838383838,0.848484848484848,0.858585858585859,0.868686868686869,0.878787878787879,0.888888888888889,0.898989898989899,0.909090909090909,0.919191919191919,0.929292929292929,0.939393939393939,0.94949494949495,0.95959595959596,0.96969696969697,0.978505606165512,null,0.0214943938344883,0.0303030303030303,0.0404040404040404,0.0505050505050505,0.0606060606060606,0.0707070707070707,0.0808080808080808,0.0909090909090909,0.101010101010101,0.111111111111111,0.121212121212121,0.131313131313131,0.141414141414141,0.151515151515151,0.161616161616162,0.171717171717172,0.181818181818182,0.191919191919192,0.202020202020202,0.212121212121212,0.222222222222222,0.232323232323232,0.242424242424242,0.252525252525253,0.262626262626263,0.272727272727273,0.282828282828283,0.292929292929293,0.303030303030303,0.313131313131313,0.323232323232323,0.333333333333333,0.343434343434343,0.353535353535354,0.363636363636364,0.373737373737374,0.383838383838384,0.393939393939394,0.404040404040404,0.414141414141414,0.424242424242424,0.434343434343434,0.444444444444444,0.454545454545455,0.464646464646465,0.474747474747475,0.484848484848485,0.494949494949495,0.505050505050505,0.515151515151515,0.525252525252525,0.535353535353535,0.545454545454546,0.555555555555556,0.565656565656566,0.575757575757576,0.585858585858586,0.595959595959596,0.606060606060606,0.616161616161616,0.626262626262626,0.636363636363636,0.646464646464647,0.656565656565656,0.666666666666667,0.676767676767677,0.686868686868687,0.696969696969697,0.707070707070707,0.717171717171717,0.727272727272727,0.737373737373737,0.747474747474748,0.757575757575758,0.767676767676768,0.777777777777778,0.787878787878788,0.797979797979798,0.808080808080808,0.818181818181818,0.828282828282828,0.838383838383838,0.848484848484848,0.858585858585859,0.868686868686869,0.878787878787879,0.888888888888889,0.898989898989899,0.909090909090909,0.919191919191919,0.929292929292929,0.939393939393939,0.94949494949495,0.95959595959596,0.96969696969697,0.978505606165512],"y":[0.151244648660193,0.156565656565657,0.164646464646465,0.172727272727273,0.180808080808081,0.188888888888889,0.196969696969697,0.205050505050505,0.213131313131313,0.221212121212121,0.229292929292929,0.237373737373737,0.245454545454545,0.253535353535354,0.261616161616162,0.26969696969697,0.277777777777778,0.285858585858586,0.293939393939394,0.302020202020202,0.31010101010101,0.318181818181818,0.326262626262626,0.334343434343434,0.342424242424242,0.350505050505051,0.358585858585859,0.366666666666667,0.374747474747475,0.382828282828283,0.390909090909091,0.398989898989899,0.407070707070707,0.415151515151515,0.423232323232323,0.431313131313131,0.439393939393939,0.447474747474747,0.455555555555556,0.463636363636364,0.471717171717172,0.47979797979798,0.487878787878788,0.495959595959596,0.504040404040404,0.512121212121212,0.52020202020202,0.528282828282828,0.536363636363636,0.544444444444444,0.552525252525253,0.560606060606061,0.568686868686869,0.576767676767677,0.584848484848485,0.592929292929293,0.601010101010101,0.609090909090909,0.617171717171717,0.625252525252525,0.633333333333333,0.641414141414141,0.649494949494949,0.657575757575757,0.665656565656566,0.673737373737374,0.681818181818182,0.68989898989899,0.697979797979798,0.706060606060606,0.714141414141414,0.722222222222222,0.73030303030303,0.738383838383838,0.746464646464646,0.754545454545454,0.762626262626263,0.770707070707071,0.778787878787879,0.786868686868687,0.794949494949495,0.803030303030303,0.811111111111111,0.819191919191919,0.827272727272727,0.835353535353535,0.843434343434343,0.851515151515151,0.859595959595959,0.867676767676768,0.875757575757576,0.883838383838384,0.891919191919192,0.893750254825705,null,0.151244648660193,0.156565656565657,0.164646464646465,0.172727272727273,0.180808080808081,0.188888888888889,0.196969696969697,0.205050505050505,0.213131313131313,0.221212121212121,0.229292929292929,0.237373737373737,0.245454545454545,0.253535353535354,0.261616161616162,0.26969696969697,0.277777777777778,0.285858585858586,0.293939393939394,0.302020202020202,0.31010101010101,0.318181818181818,0.326262626262626,0.334343434343434,0.342424242424242,0.350505050505051,0.358585858585859,0.366666666666667,0.374747474747475,0.382828282828283,0.390909090909091,0.398989898989899,0.407070707070707,0.415151515151515,0.423232323232323,0.431313131313131,0.439393939393939,0.447474747474747,0.455555555555556,0.463636363636364,0.471717171717172,0.47979797979798,0.487878787878788,0.495959595959596,0.504040404040404,0.512121212121212,0.52020202020202,0.528282828282828,0.536363636363636,0.544444444444444,0.552525252525253,0.560606060606061,0.568686868686869,0.576767676767677,0.584848484848485,0.592929292929293,0.601010101010101,0.609090909090909,0.617171717171717,0.625252525252525,0.633333333333333,0.641414141414141,0.649494949494949,0.657575757575757,0.665656565656566,0.673737373737374,0.681818181818182,0.68989898989899,0.697979797979798,0.706060606060606,0.714141414141414,0.722222222222222,0.73030303030303,0.738383838383838,0.746464646464646,0.754545454545454,0.762626262626263,0.770707070707071,0.778787878787879,0.786868686868687,0.794949494949495,0.803030303030303,0.811111111111111,0.819191919191919,0.827272727272727,0.835353535353535,0.843434343434343,0.851515151515151,0.859595959595959,0.867676767676768,0.875757575757576,0.883838383838384,0.891919191919192,0.893750254825705,null,0.151244648660193,0.156565656565657,0.164646464646465,0.172727272727273,0.180808080808081,0.188888888888889,0.196969696969697,0.205050505050505,0.213131313131313,0.221212121212121,0.229292929292929,0.237373737373737,0.245454545454545,0.253535353535354,0.261616161616162,0.26969696969697,0.277777777777778,0.285858585858586,0.293939393939394,0.302020202020202,0.31010101010101,0.318181818181818,0.326262626262626,0.334343434343434,0.342424242424242,0.350505050505051,0.358585858585859,0.366666666666667,0.374747474747475,0.382828282828283,0.390909090909091,0.398989898989899,0.407070707070707,0.415151515151515,0.423232323232323,0.431313131313131,0.439393939393939,0.447474747474747,0.455555555555556,0.463636363636364,0.471717171717172,0.47979797979798,0.487878787878788,0.495959595959596,0.504040404040404,0.512121212121212,0.52020202020202,0.528282828282828,0.536363636363636,0.544444444444444,0.552525252525253,0.560606060606061,0.568686868686869,0.576767676767677,0.584848484848485,0.592929292929293,0.601010101010101,0.609090909090909,0.617171717171717,0.625252525252525,0.633333333333333,0.641414141414141,0.649494949494949,0.657575757575757,0.665656565656566,0.673737373737374,0.681818181818182,0.68989898989899,0.697979797979798,0.706060606060606,0.714141414141414,0.722222222222222,0.73030303030303,0.738383838383838,0.746464646464646,0.754545454545454,0.762626262626263,0.770707070707071,0.778787878787879,0.786868686868687,0.794949494949495,0.803030303030303,0.811111111111111,0.819191919191919,0.827272727272727,0.835353535353535,0.843434343434343,0.851515151515151,0.859595959595959,0.867676767676768,0.875757575757576,0.883838383838384,0.891919191919192,0.893750254825705,null,0.151244648660193,0.156565656565657,0.164646464646465,0.172727272727273,0.180808080808081,0.188888888888889,0.196969696969697,0.205050505050505,0.213131313131313,0.221212121212121,0.229292929292929,0.237373737373737,0.245454545454545,0.253535353535354,0.261616161616162,0.26969696969697,0.277777777777778,0.285858585858586,0.293939393939394,0.302020202020202,0.31010101010101,0.318181818181818,0.326262626262626,0.334343434343434,0.342424242424242,0.350505050505051,0.358585858585859,0.366666666666667,0.374747474747475,0.382828282828283,0.390909090909091,0.398989898989899,0.407070707070707,0.415151515151515,0.423232323232323,0.431313131313131,0.439393939393939,0.447474747474747,0.455555555555556,0.463636363636364,0.471717171717172,0.47979797979798,0.487878787878788,0.495959595959596,0.504040404040404,0.512121212121212,0.52020202020202,0.528282828282828,0.536363636363636,0.544444444444444,0.552525252525253,0.560606060606061,0.568686868686869,0.576767676767677,0.584848484848485,0.592929292929293,0.601010101010101,0.609090909090909,0.617171717171717,0.625252525252525,0.633333333333333,0.641414141414141,0.649494949494949,0.657575757575757,0.665656565656566,0.673737373737374,0.681818181818182,0.68989898989899,0.697979797979798,0.706060606060606,0.714141414141414,0.722222222222222,0.73030303030303,0.738383838383838,0.746464646464646,0.754545454545454,0.762626262626263,0.770707070707071,0.778787878787879,0.786868686868687,0.794949494949495,0.803030303030303,0.811111111111111,0.819191919191919,0.827272727272727,0.835353535353535,0.843434343434343,0.851515151515151,0.859595959595959,0.867676767676768,0.875757575757576,0.883838383838384,0.891919191919192,0.893750254825705,null,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,null,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,null,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,null,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,null,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,null,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,null,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8],"type":"scatter","mode":"lines","line":{"width":null,"color":"transparent","dash":null},"hoveron":"points","hoverinfo":"none","showlegend":false,"_isGraticule":true,"xaxis":"x","yaxis":"y","frame":null},{"x":[0.49661783165299,0.48053143373849,0.464521869360683,0.448652320956093,0.432985418382166,0.417582991745835,0.402505827388333,0.387813427989279,0.373563777736782,0.365225932900163,0.364994903485898,0.365262147362104,0.36665312145832,0.369162336235264,0.372779888968564,0.377491502830283,0.383278583233042,0.390118291214355,0.397983633571585,0.399051104552776,0.409683140632133,0.425788737824325,0.442187212964424,0.458813848762802,0.47560302748395,0.492488489909703,0.498652466931534,0.49661783165299,0.49661783165299],"y":[0.374675028273062,0.364638714981064,0.355412868860389,0.347033900110788,0.339534876693927,0.332945393829256,0.327291457195193,0.322595380296594,0.318875696403546,0.31722117741565,0.321255419413441,0.336816851470545,0.352903896569301,0.369453066492178,0.386399049226085,0.403674966719055,0.42121263881736,0.438942852341428,0.456795634238643,0.458952868175387,0.44860341912475,0.433751952271375,0.419438194216573,0.405718634824051,0.392647418926863,0.380276132642483,0.37604080467304,0.374675028273062,0.374675028273062],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(19,43,67,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.689723867357517,0.677352581073137,0.664281365175949,0.650561805783427,0.636248047728625,0.62139658087525,0.606066017177982,0.597370908578246,0.609615927001734,0.627413129604112,0.645299472264983,0.663204365761357,0.681057147658572,0.69878736118264,0.716325033280945,0.733600950773915,0.750546933507822,0.755576593155291,0.756004420321286,0.75629779737109,0.755579684105997,0.753852914590988,0.751124303596454,0.747404619703406,0.742708542804807,0.737054606170744,0.730465123306073,0.722966099889212,0.714587131139611,0.705361285018936,0.695324971726938,0.69395919532696,0.689723867357517,0.689723867357517,null,0.593156430608217,0.583336867444015,0.572596430265486,0.560977506671335,0.548525951244807,0.535290904586595,0.521324599379342,0.506682154249102,0.5,0.50857864376269,0.52439756725828,0.540712186173608,0.557458114161256,0.574569262693944,0.591978101885752,0.597370908578246,0.606066017177982,0.62139658087525,0.636248047728625,0.650561805783427,0.664281365175949,0.677352581073137,0.689723867357517,0.69395919532696,0.684517800007025,0.672982420828649,0.660764359063413,0.647911833819203,0.634475568141398,0.620508588832182,0.606066017177982,0.600946179932138,0.593156430608217,0.593156430608217,null,0.359813113490343,0.34661570283966,0.33402362993525,0.322086589936094,0.310851692885556,0.300363277789556,0.290662737630778,0.281788356009473,0.273775156055582,0.266654762208439,0.260455275409561,0.255201162201066,0.250913158167413,0.24760818610152,0.245299289218232,0.243995579678714,0.24370220262891,0.244420315894003,0.246147085409012,0.248875696403546,0.252595380296594,0.257291457195193,0.262945393829256,0.269534876693927,0.277033900110788,0.285412868860389,0.294638714981064,0.304675028273062,0.30604080467304,0.310276132642483,0.322647418926863,0.335718634824051,0.349438194216573,0.363751952271375,0.37860341912475,0.393933982822018,0.399051104552776,0.397983633571585,0.390118291214355,0.383278583233042,0.377491502830283,0.372779888968564,0.369162336235264,0.36665312145832,0.365262147362104,0.364994903485898,0.365225932900163,0.359813113490343,0.359813113490343,null,0.471952733149137,0.458327123257497,0.445457958643352,0.433396028020072,0.422188934317255,0.411880906813711,0.402512626584708,0.39412106595235,0.386739342572712,0.380396588735586,0.375117836392638,0.370923918367733,0.367831386139297,0.365852444519191,0.365225932900163,0.373563777736782,0.387813427989279,0.402505827388333,0.417582991745835,0.432985418382166,0.448652320956093,0.464521869360683,0.48053143373849,0.49661783165299,0.498652466931534,0.5094035968341,0.526281592057887,0.543055865844739,0.559660217799456,0.576029118130679,0.592097966267041,0.607803345806105,0.623083274789937,0.634273105328381,0.634147555480809,0.632168613860703,0.629076081632267,0.624882163607362,0.619603411264414,0.613260657427288,0.60587893404765,0.597487373415292,0.588119093186289,0.577811065682745,0.566603971979928,0.554542041356648,0.541672876742503,0.528047266850863,0.513718985739022,0.5,0.486281014260978,0.471952733149137,0.471952733149137,null,0.591978101885752,0.574569262693944,0.557458114161256,0.540712186173608,0.52439756725828,0.50857864376269,0.5,0.49142135623731,0.47560243274172,0.459287813826392,0.442541885838744,0.425430737306056,0.408021898114248,0.402637560262697,0.408795148589065,0.424016258237098,0.439537241000199,0.455296842650266,0.471232867239462,0.487282422559211,0.501347533068466,0.507511510090297,0.52439697251605,0.541186151237198,0.557812787035576,0.574211262175675,0.590316859367867,0.597370908578246,0.591978101885752,0.591978101885752],"y":[0.577511510090297,0.59439697251605,0.611186151237198,0.627812787035576,0.644211262175675,0.660316859367867,0.676066017177982,0.684530020005949,0.692596430265486,0.703336867444015,0.713156430608217,0.722016366428415,0.729881708785646,0.736721416766958,0.742508497169717,0.747220111031436,0.750837663764735,0.751600269766683,0.747913410063906,0.73597637006475,0.72338429716034,0.710186886509658,0.696436222263218,0.682186572010721,0.667494172611667,0.652417008254165,0.637014581617834,0.621347679043907,0.605478130639317,0.58946856626151,0.57338216834701,0.571347533068466,0.577511510090297,0.577511510090297,null,0.474700527735017,0.492586870395888,0.510384072998266,0.528021898114248,0.545430737306056,0.562541885838744,0.579287813826392,0.59560243274172,0.602528970054552,0.611421356237309,0.626682154249102,0.641324599379342,0.655290904586595,0.668525951244807,0.680977506671335,0.684530020005949,0.676066017177982,0.660316859367867,0.644211262175675,0.627812787035576,0.611186151237198,0.59439697251605,0.577511510090297,0.571347533068466,0.557282422559211,0.541232867239462,0.525296842650266,0.509537241000199,0.494016258237098,0.478795148589065,0.463933982822018,0.458958355913603,0.474700527735017,0.474700527735017,null,0.316147085409012,0.314420315894003,0.31370220262891,0.313995579678714,0.315299289218232,0.31760818610152,0.320913158167413,0.325201162201066,0.330455275409561,0.336654762208439,0.343775156055582,0.351788356009473,0.360662737630778,0.370363277789556,0.380851692885556,0.392086589936094,0.40402362993525,0.41661570283966,0.429813113490343,0.443563777736782,0.457813427989279,0.472505827388333,0.487582991745835,0.502985418382166,0.518652320956093,0.534521869360683,0.55053143373849,0.56661783165299,0.568652466931534,0.562488489909703,0.54560302748395,0.528813848762801,0.512187212964423,0.495788737824325,0.479683140632133,0.463933982822018,0.458952868175387,0.456795634238643,0.438942852341428,0.42121263881736,0.403674966719055,0.386399049226085,0.369453066492178,0.352903896569301,0.336816851470545,0.321255419413441,0.31722117741565,0.316147085409012,0.316147085409012,null,0.187831386139297,0.190923918367733,0.195117836392638,0.200396588735586,0.206739342572712,0.21412106595235,0.222512626584708,0.231880906813711,0.242188934317255,0.253396028020072,0.265457958643352,0.278327123257497,0.291952733149137,0.306281014260978,0.31722117741565,0.318875696403546,0.322595380296594,0.327291457195193,0.332945393829256,0.339534876693927,0.347033900110788,0.355412868860389,0.364638714981064,0.374675028273062,0.37604080467304,0.368653599785973,0.357825689184687,0.347835133654967,0.338721361355239,0.330520340181081,0.32326443581638,0.316982284000754,0.311698677517379,0.308473369158742,0.306281014260978,0.291952733149137,0.278327123257497,0.265457958643352,0.253396028020072,0.242188934317255,0.231880906813711,0.222512626584708,0.21412106595235,0.206739342572712,0.200396588735586,0.195117836392638,0.190923918367733,0.187831386139297,0.185852444519191,0.185066797738128,0.185852444519191,0.187831386139297,0.187831386139297,null,0.680977506671335,0.668525951244807,0.655290904586595,0.641324599379342,0.626682154249102,0.611421356237309,0.602528970054552,0.611421356237309,0.626682154249103,0.641324599379341,0.655290904586595,0.668525951244807,0.680977506671335,0.684524441153954,0.690508588832182,0.704475568141398,0.717911833819203,0.730764359063413,0.742982420828649,0.754517800007025,0.76395919532696,0.759723867357517,0.747352581073137,0.734281365175949,0.720561805783428,0.706248047728625,0.69139658087525,0.684530020005949,0.680977506671335,0.680977506671335],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(22,50,76,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.393933982822018,0.37860341912475,0.363751952271375,0.349438194216573,0.335718634824051,0.322647418926863,0.310276132642483,0.30604080467304,0.315482199992975,0.327017579171351,0.339235640936587,0.352088166180797,0.365524431858602,0.379491411167818,0.393933982822018,0.402637560262697,0.408021898114248,0.425430737306056,0.442541885838744,0.459287813826392,0.47560243274172,0.49142135623731,0.5,0.493317845750898,0.478675400620658,0.464709095413405,0.451474048755193,0.439022493328665,0.427403569734514,0.416663132555985,0.406843569391783,0.399051104552776,0.393933982822018,0.393933982822018,null,0.602016366428415,0.609881708785646,0.616721416766958,0.622508497169717,0.627220111031436,0.630837663764735,0.63334687854168,0.634737852637896,0.635005096514101,0.634273105328381,0.623083274789937,0.607803345806105,0.592097966267041,0.576029118130679,0.559660217799456,0.543055865844739,0.526281592057887,0.5094035968341,0.498652466931534,0.512717577440789,0.528767132760538,0.544703157349734,0.560462758999801,0.575983741762902,0.591204851410935,0.600946179932138,0.602016366428415,0.602016366428415,null,0.50338216834701,0.51946856626151,0.535478130639317,0.551347679043907,0.567014581617834,0.582417008254165,0.597494172611667,0.612186572010721,0.626436222263218,0.640186886509658,0.65338429716034,0.66597637006475,0.677913410063906,0.689148307114444,0.699636722210444,0.709337262369222,0.718211643990527,0.726224843944418,0.733345237791561,0.739544724590439,0.744798837798934,0.749086841832587,0.75239181389848,0.754700710781767,0.755576593155291,0.750546933507822,0.733600950773915,0.716325033280945,0.69878736118264,0.681057147658572,0.663204365761357,0.645299472264983,0.627413129604112,0.609615927001734,0.597370908578246,0.590316859367867,0.574211262175675,0.557812787035576,0.541186151237198,0.52439697251605,0.507511510090297,0.501347533068466,0.50338216834701,0.50338216834701],"y":[0.463933982822018,0.479683140632133,0.495788737824325,0.512187212964423,0.528813848762801,0.54560302748395,0.562488489909703,0.568652466931534,0.582717577440789,0.598767132760538,0.614703157349734,0.630462758999801,0.645983741762902,0.661204851410935,0.676066017177982,0.684524441153954,0.680977506671335,0.668525951244807,0.655290904586595,0.641324599379341,0.626682154249103,0.611421356237309,0.602528970054552,0.59560243274172,0.579287813826392,0.562541885838744,0.545430737306056,0.528021898114248,0.510384072998266,0.492586870395888,0.474700527735017,0.458952868175387,0.463933982822018,0.463933982822018,null,0.456795634238643,0.438942852341428,0.42121263881736,0.403674966719055,0.386399049226085,0.369453066492178,0.352903896569301,0.336816851470545,0.321255419413441,0.308473369158742,0.311698677517379,0.316982284000754,0.32326443581638,0.330520340181081,0.338721361355239,0.347835133654967,0.357825689184687,0.368653599785973,0.37604080467304,0.385482199992975,0.397017579171351,0.409235640936587,0.422088166180797,0.435524431858602,0.449491411167818,0.458958355913603,0.456795634238643,0.456795634238643,null,0.765324971726938,0.775361285018936,0.78458713113961,0.792966099889212,0.800465123306073,0.807054606170744,0.812708542804807,0.817404619703405,0.821124303596454,0.823852914590988,0.825579684105997,0.82629779737109,0.826004420321286,0.824700710781767,0.82239181389848,0.819086841832587,0.814798837798933,0.809544724590439,0.803345237791561,0.796224843944418,0.788211643990526,0.779337262369222,0.769636722210444,0.759148307114444,0.751600269766683,0.750837663764735,0.747220111031436,0.742508497169717,0.736721416766958,0.729881708785646,0.722016366428415,0.713156430608217,0.703336867444015,0.692596430265486,0.684530020005949,0.69139658087525,0.706248047728625,0.720561805783428,0.734281365175949,0.747352581073137,0.759723867357517,0.76395919532696,0.765324971726938,0.765324971726938],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(26,56,85,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.298653599785973,0.287825689184687,0.277835133654967,0.268721361355239,0.260520340181081,0.25326443581638,0.246982284000754,0.241698677517379,0.237434468347193,0.234206485375659,0.232027467976841,0.230906015736915,0.230883038725603,0.232903896569301,0.249453066492178,0.266399049226085,0.283674966719055,0.30121263881736,0.318942852341428,0.336795634238643,0.354700527735017,0.372586870395888,0.390384072998266,0.402637560262697,0.393933982822018,0.379491411167818,0.365524431858602,0.352088166180797,0.339235640936587,0.327017579171351,0.315482199992975,0.30604080467304,0.298653599785973,0.298653599785973],"y":[0.5794035968341,0.596281592057887,0.613055865844739,0.629660217799456,0.646029118130679,0.662097966267041,0.677803345806105,0.693083274789937,0.707877450319571,0.72212748654301,0.735777145077502,0.748772556956745,0.753521612987656,0.75334687854168,0.750837663764736,0.747220111031436,0.742508497169717,0.736721416766958,0.729881708785646,0.722016366428415,0.713156430608217,0.703336867444015,0.692596430265486,0.684524441153954,0.676066017177982,0.661204851410935,0.645983741762902,0.630462758999801,0.614703157349734,0.598767132760538,0.582717577440789,0.568652466931534,0.5794035968341,0.5794035968341],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(29,63,94,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.0649949034858985,0.0658524445191911,0.0678313861392967,0.0709239183677329,0.0751178363926377,0.0803965887355857,0.0867393425727119,0.0941210659523497,0.102512626584708,0.111880906813711,0.122188934317255,0.133396028020072,0.145457958643352,0.158327123257497,0.171952733149137,0.186281014260978,0.201255419413441,0.216816851470545,0.230883038725603,0.230906015736915,0.232027467976841,0.234206485375659,0.237434468347193,0.241698677517379,0.246982284000754,0.25326443581638,0.260520340181081,0.268721361355239,0.277835133654967,0.287825689184687,0.298653599785973,0.30604080467304,0.304675028273062,0.294638714981064,0.285412868860389,0.277033900110788,0.269534876693927,0.262945393829256,0.257291457195193,0.252595380296594,0.248875696403546,0.246147085409012,0.244420315894003,0.24370220262891,0.243995579678714,0.245299289218232,0.24760818610152,0.250913158167413,0.255201162201066,0.260455275409561,0.266654762208439,0.273775156055582,0.281788356009473,0.290662737630778,0.300363277789556,0.310851692885556,0.322086589936094,0.33402362993525,0.34661570283966,0.359813113490343,0.365225932900163,0.365852444519191,0.367831386139297,0.370923918367733,0.375117836392638,0.380396588735586,0.386739342572712,0.39412106595235,0.402512626584708,0.411880906813711,0.422188934317255,0.433396028020072,0.445457958643352,0.458327123257497,0.471952733149137,0.486281014260978,0.5,0.498744580586559,0.483183148529455,0.467096103430699,0.450546933507822,0.433600950773915,0.416325033280945,0.39878736118264,0.381057147658572,0.363204365761357,0.345299472264983,0.327413129604112,0.309615927001734,0.291978101885752,0.274569262693944,0.257458114161256,0.240712186173608,0.22439756725828,0.208578643762691,0.193317845750898,0.178675400620658,0.164709095413405,0.151474048755193,0.139022493328665,0.127403569734514,0.116663132555985,0.106843569391783,0.0979836335715852,0.0901182912143544,0.0832785832330416,0.0774915028302834,0.0727798889685642,0.0691623362352644,0.0666531214583198,0.0652621473621038,0.0649949034858985,0.0649949034858985],"y":[0.618744580586559,0.633718985739022,0.648047266850863,0.661672876742503,0.674542041356648,0.686603971979928,0.697811065682745,0.708119093186289,0.717487373415292,0.72587893404765,0.733260657427288,0.739603411264414,0.744882163607362,0.749076081632267,0.752168613860703,0.754147555480809,0.755005096514102,0.754737852637896,0.753521612987656,0.748772556956745,0.735777145077502,0.72212748654301,0.707877450319571,0.693083274789937,0.677803345806105,0.662097966267041,0.646029118130679,0.629660217799456,0.613055865844739,0.596281592057887,0.5794035968341,0.568652466931534,0.56661783165299,0.55053143373849,0.534521869360683,0.518652320956093,0.502985418382166,0.487582991745835,0.472505827388333,0.457813427989279,0.443563777736782,0.429813113490343,0.41661570283966,0.40402362993525,0.392086589936094,0.380851692885556,0.370363277789556,0.360662737630778,0.351788356009473,0.343775156055582,0.336654762208439,0.330455275409561,0.325201162201066,0.320913158167413,0.31760818610152,0.315299289218232,0.313995579678714,0.31370220262891,0.314420315894003,0.316147085409012,0.31722117741565,0.306281014260978,0.291952733149137,0.278327123257497,0.265457958643352,0.253396028020072,0.242188934317255,0.231880906813711,0.222512626584708,0.21412106595235,0.206739342572712,0.200396588735586,0.195117836392638,0.190923918367733,0.187831386139297,0.185852444519191,0.185066797738128,0.184994903485898,0.185262147362104,0.18665312145832,0.189162336235264,0.192779888968564,0.197491502830283,0.203278583233042,0.210118291214354,0.217983633571585,0.226843569391783,0.236663132555985,0.247403569734514,0.259022493328665,0.271474048755193,0.284709095413405,0.298675400620658,0.313317845750897,0.32857864376269,0.34439756725828,0.360712186173608,0.377458114161256,0.394569262693944,0.411978101885752,0.429615927001734,0.447413129604112,0.465299472264983,0.483204365761357,0.501057147658572,0.518787361182639,0.536325033280944,0.553600950773915,0.570546933507822,0.587096103430699,0.603183148529455,0.618744580586559,0.618744580586559],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(33,70,103,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.406843569391783,0.416663132555985,0.427403569734514,0.439022493328665,0.451474048755193,0.464709095413405,0.478675400620658,0.493317845750898,0.5,0.506682154249102,0.521324599379342,0.535290904586595,0.548525951244807,0.560977506671335,0.572596430265486,0.583336867444015,0.593156430608217,0.600946179932138,0.591204851410935,0.575983741762902,0.560462758999801,0.544703157349734,0.528767132760538,0.512717577440789,0.498652466931534,0.492488489909703,0.47560302748395,0.458813848762802,0.442187212964424,0.425788737824325,0.409683140632133,0.399051104552776,0.406843569391783,0.406843569391783],"y":[0.474700527735017,0.492586870395888,0.510384072998266,0.528021898114248,0.545430737306056,0.562541885838744,0.579287813826392,0.59560243274172,0.602528970054552,0.59560243274172,0.579287813826392,0.562541885838744,0.545430737306056,0.528021898114248,0.510384072998266,0.492586870395888,0.474700527735017,0.458958355913603,0.449491411167818,0.435524431858602,0.422088166180797,0.409235640936587,0.397017579171351,0.385482199992975,0.37604080467304,0.380276132642483,0.392647418926863,0.405718634824051,0.419438194216573,0.433751952271375,0.44860341912475,0.458952868175387,0.474700527735017,0.474700527735017],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(36,77,113,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.500000000000001,0.513718985739022,0.528047266850863,0.541672876742503,0.554542041356648,0.566603971979928,0.577811065682745,0.588119093186289,0.597487373415292,0.60587893404765,0.613260657427288,0.619603411264414,0.624882163607362,0.629076081632267,0.632168613860703,0.634147555480809,0.634273105328381,0.637877450319571,0.65212748654301,0.665777145077502,0.678772556956745,0.691062435227093,0.702598277353723,0.713334556637987,0.723228901890489,0.732242264650813,0.740339073293959,0.747487373415292,0.75365895393998,0.75882945845922,0.762978481353853,0.766089648326023,0.768150681021047,0.769153445484473,0.769093984263086,0.767972532023159,0.765793514624341,0.762565531652807,0.758301322482621,0.753017715999246,0.74673556418362,0.739479659818919,0.731278638644761,0.722164866345033,0.712174310815313,0.701346400214027,0.69395919532696,0.695324971726938,0.705361285018936,0.714587131139611,0.722966099889212,0.730465123306073,0.737054606170744,0.742708542804807,0.747404619703406,0.751124303596454,0.753852914590988,0.755579684105997,0.75629779737109,0.756004420321286,0.755576593155291,0.767096103430699,0.783183148529455,0.798744580586559,0.813718985739022,0.828047266850863,0.841672876742503,0.854542041356648,0.866603971979928,0.877811065682745,0.888119093186289,0.897487373415292,0.90587893404765,0.913260657427288,0.919603411264414,0.924882163607362,0.929076081632267,0.932168613860703,0.934147555480809,0.935005096514102,0.934737852637896,0.93334687854168,0.930837663764736,0.927220111031436,0.922508497169717,0.916721416766959,0.909881708785646,0.902016366428415,0.893156430608217,0.883336867444015,0.872596430265486,0.860977506671335,0.848525951244807,0.835290904586595,0.821324599379342,0.806682154249103,0.79142135623731,0.77560243274172,0.759287813826392,0.742541885838744,0.725430737306056,0.708021898114248,0.690384072998266,0.672586870395888,0.654700527735017,0.636795634238643,0.618942852341428,0.60121263881736,0.583674966719055,0.566399049226085,0.549453066492178,0.532903896569301,0.516816851470545,0.501255419413441,0.500000000000001,0.500000000000001],"y":[0.185066797738128,0.185852444519191,0.187831386139297,0.190923918367733,0.195117836392638,0.200396588735586,0.206739342572712,0.21412106595235,0.222512626584708,0.231880906813711,0.242188934317255,0.253396028020072,0.265457958643352,0.278327123257497,0.291952733149137,0.306281014260978,0.308473369158742,0.307434468347192,0.304206485375659,0.302027467976841,0.300906015736914,0.300846554515527,0.301849318978953,0.303910351673977,0.307021518646146,0.31117054154078,0.31634104606002,0.322512626584708,0.329660926706041,0.337757735349187,0.346771098109511,0.356665443362013,0.367401722646276,0.378937564772907,0.391227443043255,0.404222854922498,0.41787251345699,0.432122549680429,0.446916725210063,0.462196654193895,0.477902033732959,0.493970881869321,0.510339782200544,0.526944134155261,0.543718407942113,0.5605964031659,0.571347533068466,0.57338216834701,0.58946856626151,0.605478130639317,0.621347679043907,0.637014581617834,0.652417008254165,0.667494172611667,0.682186572010721,0.696436222263218,0.710186886509658,0.72338429716034,0.73597637006475,0.747913410063906,0.751600269766683,0.75334687854168,0.754737852637896,0.755005096514101,0.754147555480809,0.752168613860703,0.749076081632267,0.744882163607362,0.739603411264414,0.733260657427288,0.72587893404765,0.717487373415292,0.708119093186289,0.697811065682745,0.686603971979928,0.674542041356648,0.661672876742503,0.648047266850863,0.633718985739022,0.618744580586559,0.603183148529455,0.587096103430699,0.570546933507822,0.553600950773915,0.536325033280944,0.51878736118264,0.501057147658572,0.483204365761357,0.465299472264983,0.447413129604112,0.429615927001734,0.411978101885752,0.394569262693944,0.377458114161256,0.360712186173608,0.34439756725828,0.32857864376269,0.313317845750897,0.298675400620658,0.284709095413405,0.271474048755193,0.259022493328665,0.247403569734514,0.236663132555985,0.226843569391783,0.217983633571585,0.210118291214354,0.203278583233042,0.197491502830283,0.192779888968564,0.189162336235264,0.18665312145832,0.185262147362104,0.184994903485898,0.185066797738128,0.185066797738128],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(55,114,162,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.635005096514101,0.634737852637896,0.63334687854168,0.630837663764735,0.627220111031436,0.622508497169717,0.616721416766958,0.609881708785646,0.602016366428415,0.600946179932138,0.606066017177982,0.620508588832182,0.634475568141398,0.647911833819203,0.660764359063413,0.672982420828649,0.684517800007025,0.69395919532696,0.701346400214027,0.712174310815313,0.722164866345033,0.731278638644761,0.739479659818919,0.74673556418362,0.753017715999246,0.758301322482621,0.762565531652807,0.765793514624341,0.767972532023159,0.769093984263086,0.769153445484473,0.768150681021047,0.766089648326023,0.762978481353853,0.75882945845922,0.75365895393998,0.747487373415292,0.740339073293959,0.732242264650813,0.723228901890489,0.713334556637987,0.702598277353723,0.691062435227093,0.678772556956745,0.665777145077502,0.65212748654301,0.637877450319571,0.634273105328381,0.635005096514101,0.635005096514101],"y":[0.321255419413441,0.336816851470545,0.352903896569301,0.369453066492178,0.386399049226085,0.403674966719055,0.42121263881736,0.438942852341428,0.456795634238643,0.458958355913603,0.463933982822018,0.478795148589065,0.494016258237098,0.509537241000199,0.525296842650266,0.541232867239462,0.557282422559211,0.571347533068466,0.5605964031659,0.543718407942113,0.526944134155261,0.510339782200544,0.493970881869321,0.477902033732959,0.462196654193895,0.446916725210063,0.432122549680429,0.41787251345699,0.404222854922498,0.391227443043255,0.378937564772907,0.367401722646276,0.356665443362013,0.346771098109511,0.337757735349187,0.329660926706041,0.322512626584708,0.31634104606002,0.31117054154078,0.307021518646146,0.303910351673977,0.301849318978953,0.300846554515527,0.300906015736914,0.302027467976841,0.304206485375659,0.307434468347192,0.308473369158742,0.321255419413441,0.321255419413441],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(62,129,183,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.487282422559211,0.471232867239462,0.455296842650266,0.439537241000199,0.424016258237098,0.408795148589065,0.402637560262697,0.390384072998266,0.372586870395888,0.354700527735017,0.336795634238643,0.318942852341428,0.30121263881736,0.283674966719055,0.266399049226085,0.249453066492178,0.232903896569301,0.230883038725603,0.230846554515527,0.231849318978953,0.233910351673977,0.237021518646147,0.24117054154078,0.24634104606002,0.252512626584708,0.259660926706041,0.267757735349187,0.276771098109511,0.286665443362013,0.297401722646276,0.308937564772907,0.321227443043255,0.334222854922498,0.34787251345699,0.362122549680428,0.376916725210063,0.392196654193895,0.407902033732959,0.423970881869321,0.440339782200544,0.456944134155261,0.473718407942113,0.4905964031659,0.501347533068466,0.487282422559211,0.487282422559211],"y":[0.754517800007025,0.742982420828649,0.730764359063413,0.717911833819203,0.704475568141398,0.690508588832182,0.684524441153954,0.692596430265486,0.703336867444015,0.713156430608217,0.722016366428415,0.729881708785646,0.736721416766958,0.742508497169717,0.747220111031436,0.750837663764736,0.75334687854168,0.753521612987656,0.761062435227093,0.772598277353723,0.783334556637987,0.793228901890489,0.802242264650813,0.810339073293959,0.817487373415292,0.82365895393998,0.82882945845922,0.832978481353853,0.836089648326023,0.838150681021047,0.839153445484473,0.839093984263085,0.837972532023159,0.835793514624341,0.832565531652807,0.828301322482621,0.823017715999246,0.81673556418362,0.809479659818919,0.801278638644761,0.792164866345032,0.782174310815313,0.771346400214027,0.76395919532696,0.754517800007025,0.754517800007025],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(86,177,247,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.08,0.26,0.71,0.93],"y":[0.78,0.86,0.85,0.78],"text":["eco-activated","eco-suppressed","ppu-activated","ppu-suppressed"],"hovertext":["","","",""],"textfont":{"size":14.6645669291339,"color":"rgba(0,0,0,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.20045299544466,0.358381732758067,0.633396116298188,0.802485690816114,0.306472730239411,0.315123952707465,0.498421428685342,0.500121592161113,0.690655684959883,0.689238063246841,0.39953102161938,0.582377337297941,0.417284378929032,0.600468975765232,0.499924063415392],"y":[0.519090468430361,0.771384232274224,0.765202418425565,0.529878786855597,0.674019656966952,0.425613540671737,0.273819561432775,0.6863829572591,0.418131940967582,0.673276962924277,0.579085938438095,0.377764551702095,0.380690624900884,0.579259049632578,0.481316460773137],"text":[4,18,2,10,3,1,1,1,12,1,2,2,0,1,5],"hovertext":["Arginine biosynthesis Biosynthesis<br />of amino acids 2-Oxocarboxylic<br />acid metabolism Valine, leucine and<br />isoleucine biosynthesis","Selenocompound metabolism Amino<br />sugar and nucleotide sugar metabolism<br />Purine metabolism Arginine and proline<br />metabolism Pentose and glucuronate<br />interconversions Ascorbate and aldarate<br />metabolism Fructose and mannose<br />metabolism Galactose metabolism<br />Butanoate metabolism Glycerophospholipid<br />metabolism Glycerolipid metabolism<br />Glutathione metabolism Alanine,<br />aspartate and glutamate metabolism<br />Fatty acid metabolism Biosynthesis of<br />cofactors beta-Lactam resistance Folate<br />biosynthesis Sulfur relay system","Porphyrin and chlorophyll metabolism<br />Protein export","Pentose phosphate pathway Methane<br />metabolism Streptomycin biosynthesis<br />Propanoate metabolism Valine, leucine<br />and isoleucine degradation Synthesis<br />and degradation of ketone bodies<br />Tyrosine metabolism Degradation of<br />aromatic compounds Glycine, serine and<br />threonine metabolism Biofilm formation -<br />Pseudomonas aeruginosa","Nitrogen metabolism Phosphotransferase<br />system (PTS) Citrate cycle (TCA cycle)","Bacterial secretion system","Lipopolysaccharide biosynthesis","Nicotinate and nicotinamide metabolism","Carbon metabolism Pyruvate metabolism<br />Glycolysis / Gluconeogenesis Glyoxylate<br />and dicarboxylate metabolism Two-<br />component system Biosynthesis of<br />secondary metabolites beta-Alanine<br />metabolism Lysine degradation Benzoate<br />degradation Tryptophan metabolism Fatty<br />acid degradation Pyrimidine metabolism","RNA degradation","Flagellar assembly Sulfur metabolism","Starch and sucrose metabolism Oxidative<br />phosphorylation","","Quorum sensing","Aminoacyl-tRNA biosynthesis ABC<br />transporters Ribosome Phenylalanine<br />metabolism Microbial metabolism in<br />diverse environments"],"textfont":{"size":14.6645669291339,"color":"rgba(0,0,0,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.2],"y":[0.2],"name":"99_c0049e8f2ad691b3655c0ba82423907b","type":"scatter","mode":"markers","opacity":0,"hoverinfo":"skip","showlegend":false,"marker":{"color":[0,1],"colorscale":[[0,"#132B43"],[0.00334448160535117,"#132B44"],[0.00668896321070234,"#132C44"],[0.0100334448160535,"#142C45"],[0.0133779264214047,"#142D45"],[0.0167224080267559,"#142D46"],[0.020066889632107,"#142D46"],[0.0234113712374582,"#142E47"],[0.0267558528428094,"#152E47"],[0.0301003344481605,"#152F48"],[0.0334448160535117,"#152F48"],[0.0367892976588629,"#152F49"],[0.040133779264214,"#153049"],[0.0434782608695652,"#16304A"],[0.0468227424749164,"#16304A"],[0.0501672240802676,"#16314B"],[0.0535117056856187,"#16314B"],[0.0568561872909699,"#16324C"],[0.0602006688963211,"#17324D"],[0.0635451505016723,"#17324D"],[0.0668896321070234,"#17334E"],[0.0702341137123746,"#17334E"],[0.0735785953177258,"#17344F"],[0.0769230769230769,"#18344F"],[0.0802675585284281,"#183450"],[0.0836120401337793,"#183550"],[0.0869565217391304,"#183551"],[0.0903010033444816,"#183651"],[0.0936454849498328,"#193652"],[0.0969899665551839,"#193652"],[0.100334448160535,"#193753"],[0.103678929765886,"#193754"],[0.107023411371237,"#193854"],[0.110367892976589,"#1A3855"],[0.11371237458194,"#1A3955"],[0.117056856187291,"#1A3956"],[0.120401337792642,"#1A3956"],[0.123745819397993,"#1A3A57"],[0.127090301003345,"#1B3A57"],[0.130434782608696,"#1B3B58"],[0.133779264214047,"#1B3B59"],[0.137123745819398,"#1B3B59"],[0.140468227424749,"#1C3C5A"],[0.1438127090301,"#1C3C5A"],[0.147157190635452,"#1C3D5B"],[0.150501672240803,"#1C3D5B"],[0.153846153846154,"#1C3D5C"],[0.157190635451505,"#1D3E5C"],[0.160535117056856,"#1D3E5D"],[0.163879598662207,"#1D3F5D"],[0.167224080267559,"#1D3F5E"],[0.17056856187291,"#1D3F5F"],[0.173913043478261,"#1E405F"],[0.177257525083612,"#1E4060"],[0.180602006688963,"#1E4160"],[0.183946488294314,"#1E4161"],[0.187290969899666,"#1E4261"],[0.190635451505017,"#1F4262"],[0.193979933110368,"#1F4263"],[0.197324414715719,"#1F4363"],[0.20066889632107,"#1F4364"],[0.204013377926421,"#1F4464"],[0.207357859531773,"#204465"],[0.210702341137124,"#204465"],[0.214046822742475,"#204566"],[0.217391304347826,"#204566"],[0.220735785953177,"#214667"],[0.224080267558528,"#214668"],[0.22742474916388,"#214768"],[0.230769230769231,"#214769"],[0.234113712374582,"#214769"],[0.237458193979933,"#22486A"],[0.240802675585284,"#22486A"],[0.244147157190635,"#22496B"],[0.247491638795987,"#22496C"],[0.250836120401338,"#224A6C"],[0.254180602006689,"#234A6D"],[0.25752508361204,"#234A6D"],[0.260869565217391,"#234B6E"],[0.264214046822742,"#234B6E"],[0.267558528428094,"#244C6F"],[0.270903010033445,"#244C70"],[0.274247491638796,"#244C70"],[0.277591973244147,"#244D71"],[0.280936454849498,"#244D71"],[0.284280936454849,"#254E72"],[0.287625418060201,"#254E72"],[0.290969899665552,"#254F73"],[0.294314381270903,"#254F74"],[0.297658862876254,"#254F74"],[0.301003344481605,"#265075"],[0.304347826086957,"#265075"],[0.307692307692308,"#265176"],[0.311036789297659,"#265176"],[0.31438127090301,"#275277"],[0.317725752508361,"#275278"],[0.321070234113712,"#275278"],[0.324414715719064,"#275379"],[0.327759197324415,"#275379"],[0.331103678929766,"#28547A"],[0.334448160535117,"#28547B"],[0.337792642140468,"#28557B"],[0.341137123745819,"#28557C"],[0.344481605351171,"#28567C"],[0.347826086956522,"#29567D"],[0.351170568561873,"#29567D"],[0.354515050167224,"#29577E"],[0.357859531772575,"#29577F"],[0.361204013377926,"#2A587F"],[0.364548494983278,"#2A5880"],[0.367892976588629,"#2A5980"],[0.37123745819398,"#2A5981"],[0.374581939799331,"#2A5982"],[0.377926421404682,"#2B5A82"],[0.381270903010033,"#2B5A83"],[0.384615384615385,"#2B5B83"],[0.387959866220736,"#2B5B84"],[0.391304347826087,"#2C5C85"],[0.394648829431438,"#2C5C85"],[0.397993311036789,"#2C5D86"],[0.40133779264214,"#2C5D86"],[0.404682274247492,"#2C5D87"],[0.408026755852843,"#2D5E87"],[0.411371237458194,"#2D5E88"],[0.414715719063545,"#2D5F89"],[0.418060200668896,"#2D5F89"],[0.421404682274247,"#2E608A"],[0.424749163879599,"#2E608A"],[0.42809364548495,"#2E618B"],[0.431438127090301,"#2E618C"],[0.434782608695652,"#2E618C"],[0.438127090301003,"#2F628D"],[0.441471571906354,"#2F628D"],[0.444816053511706,"#2F638E"],[0.448160535117057,"#2F638F"],[0.451505016722408,"#30648F"],[0.454849498327759,"#306490"],[0.45819397993311,"#306590"],[0.461538461538462,"#306591"],[0.464882943143813,"#306592"],[0.468227424749164,"#316692"],[0.471571906354515,"#316693"],[0.474916387959866,"#316793"],[0.478260869565217,"#316794"],[0.481605351170569,"#326895"],[0.48494983277592,"#326895"],[0.488294314381271,"#326996"],[0.491638795986622,"#326996"],[0.494983277591973,"#326997"],[0.498327759197324,"#336A98"],[0.501672240802676,"#336A98"],[0.505016722408027,"#336B99"],[0.508361204013378,"#336B99"],[0.511705685618729,"#346C9A"],[0.51505016722408,"#346C9B"],[0.518394648829431,"#346D9B"],[0.521739130434783,"#346D9C"],[0.525083612040134,"#346E9D"],[0.528428093645485,"#356E9D"],[0.531772575250836,"#356E9E"],[0.535117056856187,"#356F9E"],[0.538461538461539,"#356F9F"],[0.54180602006689,"#3670A0"],[0.545150501672241,"#3670A0"],[0.548494983277592,"#3671A1"],[0.551839464882943,"#3671A1"],[0.555183946488294,"#3772A2"],[0.558528428093645,"#3772A3"],[0.561872909698997,"#3773A3"],[0.565217391304348,"#3773A4"],[0.568561872909699,"#3773A4"],[0.57190635451505,"#3874A5"],[0.575250836120401,"#3874A6"],[0.578595317725753,"#3875A6"],[0.581939799331104,"#3875A7"],[0.585284280936455,"#3976A8"],[0.588628762541806,"#3976A8"],[0.591973244147157,"#3977A9"],[0.595317725752508,"#3977A9"],[0.598662207357859,"#3978AA"],[0.602006688963211,"#3A78AB"],[0.605351170568562,"#3A79AB"],[0.608695652173913,"#3A79AC"],[0.612040133779264,"#3A79AC"],[0.615384615384615,"#3B7AAD"],[0.618729096989967,"#3B7AAE"],[0.622073578595318,"#3B7BAE"],[0.625418060200669,"#3B7BAF"],[0.62876254180602,"#3C7CB0"],[0.632107023411371,"#3C7CB0"],[0.635451505016722,"#3C7DB1"],[0.638795986622074,"#3C7DB1"],[0.642140468227425,"#3C7EB2"],[0.645484949832776,"#3D7EB3"],[0.648829431438127,"#3D7FB3"],[0.652173913043478,"#3D7FB4"],[0.655518394648829,"#3D7FB5"],[0.658862876254181,"#3E80B5"],[0.662207357859532,"#3E80B6"],[0.665551839464883,"#3E81B6"],[0.668896321070234,"#3E81B7"],[0.672240802675585,"#3F82B8"],[0.675585284280936,"#3F82B8"],[0.678929765886288,"#3F83B9"],[0.682274247491639,"#3F83BA"],[0.68561872909699,"#4084BA"],[0.688963210702341,"#4084BB"],[0.692307692307692,"#4085BB"],[0.695652173913043,"#4085BC"],[0.698996655518395,"#4086BD"],[0.702341137123746,"#4186BD"],[0.705685618729097,"#4186BE"],[0.709030100334448,"#4187BF"],[0.712374581939799,"#4187BF"],[0.715719063545151,"#4288C0"],[0.719063545150502,"#4288C1"],[0.722408026755853,"#4289C1"],[0.725752508361204,"#4289C2"],[0.729096989966555,"#438AC2"],[0.732441471571906,"#438AC3"],[0.735785953177257,"#438BC4"],[0.739130434782609,"#438BC4"],[0.74247491638796,"#438CC5"],[0.745819397993311,"#448CC6"],[0.749163879598662,"#448DC6"],[0.752508361204013,"#448DC7"],[0.755852842809365,"#448EC8"],[0.759197324414716,"#458EC8"],[0.762541806020067,"#458FC9"],[0.765886287625418,"#458FC9"],[0.769230769230769,"#458FCA"],[0.77257525083612,"#4690CB"],[0.775919732441471,"#4690CB"],[0.779264214046823,"#4691CC"],[0.782608695652174,"#4691CD"],[0.785953177257525,"#4792CD"],[0.789297658862876,"#4792CE"],[0.792642140468227,"#4793CF"],[0.795986622073579,"#4793CF"],[0.79933110367893,"#4894D0"],[0.802675585284281,"#4894D0"],[0.806020066889632,"#4895D1"],[0.809364548494983,"#4895D2"],[0.812709030100334,"#4896D2"],[0.816053511705686,"#4996D3"],[0.819397993311037,"#4997D4"],[0.822742474916388,"#4997D4"],[0.826086956521739,"#4998D5"],[0.82943143812709,"#4A98D6"],[0.832775919732442,"#4A99D6"],[0.836120401337793,"#4A99D7"],[0.839464882943144,"#4A9AD8"],[0.842809364548495,"#4B9AD8"],[0.846153846153846,"#4B9BD9"],[0.849498327759197,"#4B9BDA"],[0.852842809364548,"#4B9BDA"],[0.8561872909699,"#4C9CDB"],[0.859531772575251,"#4C9CDB"],[0.862876254180602,"#4C9DDC"],[0.866220735785953,"#4C9DDD"],[0.869565217391304,"#4D9EDD"],[0.872909698996656,"#4D9EDE"],[0.876254180602007,"#4D9FDF"],[0.879598662207358,"#4D9FDF"],[0.882943143812709,"#4DA0E0"],[0.88628762541806,"#4EA0E1"],[0.889632107023411,"#4EA1E1"],[0.892976588628763,"#4EA1E2"],[0.896321070234114,"#4EA2E3"],[0.899665551839465,"#4FA2E3"],[0.903010033444816,"#4FA3E4"],[0.906354515050167,"#4FA3E5"],[0.909698996655518,"#4FA4E5"],[0.91304347826087,"#50A4E6"],[0.916387959866221,"#50A5E7"],[0.919732441471572,"#50A5E7"],[0.923076923076923,"#50A6E8"],[0.926421404682274,"#51A6E8"],[0.929765886287626,"#51A7E9"],[0.933110367892977,"#51A7EA"],[0.936454849498328,"#51A8EA"],[0.939799331103679,"#52A8EB"],[0.94314381270903,"#52A9EC"],[0.946488294314381,"#52A9EC"],[0.949832775919732,"#52AAED"],[0.953177257525084,"#53AAEE"],[0.956521739130435,"#53ABEE"],[0.959866220735786,"#53ABEF"],[0.963210702341137,"#53ACF0"],[0.966555183946488,"#54ACF0"],[0.969899665551839,"#54ADF1"],[0.973244147157191,"#54ADF2"],[0.976588628762542,"#54AEF2"],[0.979933110367893,"#55AEF3"],[0.983277591973244,"#55AFF4"],[0.986622073578595,"#55AFF4"],[0.989966555183946,"#55B0F5"],[0.993311036789298,"#56B0F6"],[0.996655518394649,"#56B1F6"],[1,"#56B1F7"]],"colorbar":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"thickness":23.04,"title":"count","titlefont":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"tickmode":"array","ticktext":["0","5","10","15"],"tickvals":[0,0.277777777777778,0.555555555555556,0.833333333333333],"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"ticklen":2,"len":0.5}},"xaxis":"x","yaxis":"y","frame":null}],"layout":{"margin":{"t":16,"r":0,"b":0,"l":0},"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.0214943938344883,0.978505606165512],"tickmode":"array","ticktext":["0.2","0.4","0.6","0.8"],"tickvals":[0.2,0.4,0.6,0.8],"categoryorder":"array","categoryarray":["0.2","0.4","0.6","0.8"],"nticks":null,"ticks":"outside","tickcolor":null,"ticklen":0,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"","font":{"color":null,"family":null,"size":0}},"scaleanchor":"y","scaleratio":1,"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.151244648660193,0.893750254825705],"tickmode":"array","ticktext":["0.2","0.3","0.4","0.5","0.6","0.7","0.8"],"tickvals":[0.2,0.3,0.4,0.5,0.6,0.7,0.8],"categoryorder":"array","categoryarray":["0.2","0.3","0.4","0.5","0.6","0.7","0.8"],"nticks":null,"ticks":"outside","tickcolor":null,"ticklen":0,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":{"text":"","font":{"color":null,"family":null,"size":0}},"scaleanchor":"x","scaleratio":1,"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":false,"legend":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"2ee025351439":{"fill":{},"type":"scatter"},"2ee051ec5d16":{"label":{}},"2ee030814730":{"label":{},"text":{}}},"cur_data":"2ee025351439","visdat":{"2ee025351439":["function (y) ","x"],"2ee051ec5d16":["function (y) ","x"],"2ee030814730":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 13: Venn plot
</p>
</div>
</div>
</div>
<div id="session-info" class="section level2">
<h2>Session info</h2>
<pre class="r"><code>sessioninfo::session_info()</code></pre>
<pre><code>## - Session info ---------------------------------------------------------------
##  setting  value                         
##  version  R version 4.1.0 (2021-05-18)  
##  os       Windows 10 x64                
##  system   x86_64, mingw32               
##  ui       RTerm                         
##  language (EN)                          
##  collate  Chinese (Simplified)_China.936
##  ctype    Chinese (Simplified)_China.936
##  tz       Asia/Taipei                   
##  date     2021-07-23                    
## 
## - Packages -------------------------------------------------------------------
##  package              * version   date       lib
##  abind                  1.4-5     2016-07-21 [1]
##  annotate               1.70.0    2021-05-19 [1]
##  AnnotationDbi          1.54.1    2021-06-08 [1]
##  ape                  * 5.5       2021-04-25 [1]
##  aplot                  0.0.6     2020-09-03 [1]
##  assertthat             0.2.1     2019-03-21 [1]
##  backports              1.2.1     2020-12-09 [1]
##  base64enc              0.1-3     2015-07-28 [1]
##  Biobase              * 2.52.0    2021-05-19 [1]
##  BiocGenerics         * 0.38.0    2021-05-19 [1]
##  BiocManager            1.30.16   2021-06-15 [1]
##  BiocParallel           1.26.1    2021-07-04 [1]
##  Biostrings             2.60.1    2021-06-06 [1]
##  bit                    4.0.4     2020-08-04 [1]
##  bit64                  4.0.5     2020-08-30 [1]
##  bitops                 1.0-7     2021-04-24 [1]
##  blob                   1.2.1     2020-01-20 [1]
##  bookdown               0.22      2021-04-22 [1]
##  broom                  0.7.8     2021-06-24 [1]
##  bslib                  0.2.5.1   2021-05-18 [1]
##  cachem                 1.0.5     2021-05-15 [1]
##  car                    3.0-11    2021-06-27 [1]
##  carData                3.0-4     2020-05-22 [1]
##  cellranger             1.1.0     2016-07-27 [1]
##  class                  7.3-19    2021-05-03 [1]
##  classInt               0.4-3     2020-04-07 [1]
##  cli                    3.0.0     2021-06-30 [1]
##  cluster                2.1.2     2021-04-17 [1]
##  clusterProfiler      * 4.0.2     2021-07-06 [1]
##  colorspace             2.0-2     2021-06-24 [1]
##  corrplot             * 0.90      2021-06-30 [1]
##  cowplot              * 1.1.1     2020-12-30 [1]
##  crayon                 1.4.1     2021-02-08 [1]
##  crosstalk              1.1.1     2021-01-12 [1]
##  curl                   4.3.2     2021-06-23 [1]
##  data.table             1.14.0    2021-02-21 [1]
##  DBI                    1.1.1     2021-01-15 [1]
##  dbplyr                 2.1.1     2021-04-06 [1]
##  DelayedArray           0.18.0    2021-05-19 [1]
##  DESeq2               * 1.32.0    2021-05-19 [1]
##  devEMF                 4.0-2     2020-10-01 [1]
##  digest                 0.6.27    2020-10-24 [1]
##  directlabels           2021.1.13 2021-01-16 [1]
##  DO.db                  2.9       2021-07-16 [1]
##  DOSE                   3.18.1    2021-06-22 [1]
##  downloader             0.4       2015-07-09 [1]
##  dplyr                * 1.0.7     2021-06-18 [1]
##  e1071                  1.7-7     2021-05-23 [1]
##  ellipsis               0.3.2     2021-04-29 [1]
##  enrichplot             1.12.2    2021-07-01 [1]
##  evaluate               0.14      2019-05-28 [1]
##  export                 0.3.0     2021-07-16 [1]
##  extrafont              0.17      2014-12-08 [1]
##  extrafontdb            1.0       2012-06-11 [1]
##  fansi                  0.5.0     2021-05-25 [1]
##  farver                 2.1.0     2021-02-28 [1]
##  fastmap                1.1.0     2021-01-25 [1]
##  fastmatch              1.1-0     2017-01-28 [1]
##  fgsea                  1.18.0    2021-05-19 [1]
##  flextable              0.6.6     2021-05-17 [1]
##  forcats              * 0.5.1     2021-01-27 [1]
##  foreign                0.8-81    2020-12-22 [1]
##  fs                     1.5.0     2020-07-31 [1]
##  gdtools              * 0.2.3     2021-01-06 [1]
##  genefilter             1.74.0    2021-05-19 [1]
##  geneplotter            1.70.0    2021-05-19 [1]
##  generics               0.1.0     2020-10-31 [1]
##  GenomeInfoDb         * 1.28.1    2021-07-01 [1]
##  GenomeInfoDbData       1.2.6     2021-07-16 [1]
##  GenomicRanges        * 1.44.0    2021-05-19 [1]
##  ggforce                0.3.3     2021-03-05 [1]
##  ggplot2              * 3.3.5     2021-06-25 [1]
##  ggpubr               * 0.4.0     2020-06-27 [1]
##  ggraph                 2.0.5     2021-02-23 [1]
##  ggrepel                0.9.1     2021-01-15 [1]
##  ggsignif               0.6.2     2021-06-14 [1]
##  ggtext               * 0.1.1     2020-12-17 [1]
##  ggtree               * 3.0.2     2021-06-01 [1]
##  ggVennDiagram        * 1.1.4     2021-07-07 [1]
##  glue                   1.4.2     2020-08-27 [1]
##  GO.db                  3.13.0    2021-07-16 [1]
##  GOSemSim               2.18.0    2021-05-19 [1]
##  graphlayouts           0.7.1     2020-10-26 [1]
##  gridExtra              2.3       2017-09-09 [1]
##  gridtext               0.1.4     2020-12-10 [1]
##  gtable               * 0.3.0     2019-03-25 [1]
##  haven                  2.4.1     2021-04-23 [1]
##  highr                  0.9       2021-04-16 [1]
##  hms                    1.1.0     2021-05-17 [1]
##  htmltools              0.5.1.1   2021-01-22 [1]
##  htmlwidgets            1.5.3     2020-12-10 [1]
##  httpuv                 1.6.1     2021-05-07 [1]
##  httr                   1.4.2     2020-07-20 [1]
##  igraph                 1.2.6     2020-10-06 [1]
##  IRanges              * 2.26.0    2021-05-19 [1]
##  jquerylib              0.1.4     2021-04-26 [1]
##  jsonlite               1.7.2     2020-12-09 [1]
##  KEGG.db                1.0       2021-03-17 [1]
##  KEGGREST               1.32.0    2021-05-19 [1]
##  KernSmooth             2.23-20   2021-05-03 [1]
##  knitr                  1.33      2021-04-24 [1]
##  labeling               0.4.2     2020-10-20 [1]
##  later                  1.2.0     2021-04-23 [1]
##  lattice              * 0.20-44   2021-05-02 [1]
##  lazyeval               0.2.2     2019-03-15 [1]
##  lifecycle              1.0.0     2021-02-15 [1]
##  locfit                 1.5-9.4   2020-03-25 [1]
##  lubridate              1.7.10    2021-02-26 [1]
##  magrittr               2.0.1     2020-11-17 [1]
##  manipulateWidget       0.11.0    2021-05-31 [1]
##  markdown               1.1       2019-08-07 [1]
##  MASS                   7.3-54    2021-05-03 [1]
##  Matrix                 1.3-4     2021-06-01 [1]
##  MatrixGenerics       * 1.4.0     2021-05-19 [1]
##  matrixStats          * 0.59.0    2021-06-01 [1]
##  memoise                2.0.0     2021-01-26 [1]
##  mgcv                   1.8-36    2021-06-01 [1]
##  mime                   0.11      2021-06-23 [1]
##  miniUI                 0.1.1.1   2018-05-18 [1]
##  modelr                 0.1.8     2020-05-19 [1]
##  munsell                0.5.0     2018-06-12 [1]
##  nlme                   3.1-152   2021-02-04 [1]
##  officer                0.3.18    2021-04-02 [1]
##  openxlsx             * 4.2.4     2021-06-16 [1]
##  patchwork              1.1.1     2020-12-17 [1]
##  permute              * 0.9-5     2019-03-12 [1]
##  pheatmap             * 1.0.12    2019-01-04 [1]
##  pillar                 1.6.1     2021-05-16 [1]
##  pkgconfig              2.0.3     2019-09-22 [1]
##  pkgdown                1.6.1     2020-09-12 [1]
##  plotly                 4.9.4.1   2021-06-18 [1]
##  plyr                   1.8.6     2020-03-03 [1]
##  png                    0.1-7     2013-12-03 [1]
##  polyclip               1.10-0    2019-03-14 [1]
##  promises               1.2.0.1   2021-02-11 [1]
##  proxy                  0.4-26    2021-06-07 [1]
##  purrr                * 0.3.4     2020-04-17 [1]
##  quadprog               1.5-8     2019-11-20 [1]
##  qvalue                 2.24.0    2021-05-19 [1]
##  R6                     2.5.0     2020-10-28 [1]
##  RColorBrewer         * 1.1-2     2014-12-07 [1]
##  Rcpp                   1.0.7     2021-07-07 [1]
##  RCurl                  1.98-1.3  2021-03-16 [1]
##  readr                * 1.4.0     2020-10-05 [1]
##  readxl                 1.3.1     2019-03-13 [1]
##  reprex                 2.0.0     2021-04-02 [1]
##  reshape2               1.4.4     2020-04-09 [1]
##  rgl                    0.106.8   2021-04-23 [1]
##  rio                    0.5.27    2021-06-21 [1]
##  rlang                  0.4.11    2021-04-30 [1]
##  rmarkdown              2.9       2021-06-15 [1]
##  RSQLite                2.2.7     2021-04-22 [1]
##  rstatix              * 0.7.0     2021-02-13 [1]
##  rstudioapi             0.13      2020-11-12 [1]
##  Rttf2pt1               1.3.8     2020-01-10 [1]
##  rvcheck                0.1.8     2020-03-01 [1]
##  RVenn                  1.1.0     2019-07-18 [1]
##  rvest                  1.0.0     2021-03-09 [1]
##  rvg                    0.2.5     2020-06-30 [1]
##  S4Vectors            * 0.30.0    2021-05-19 [1]
##  sass                   0.4.0     2021-05-12 [1]
##  scales                 1.1.1     2020-05-11 [1]
##  scatterpie             0.1.6     2021-04-23 [1]
##  sessioninfo            1.1.1     2018-11-05 [1]
##  sf                     1.0-1     2021-06-29 [1]
##  shadowtext             0.0.8     2021-04-23 [1]
##  shiny                  1.6.0     2021-01-25 [1]
##  snow                   0.4-3     2018-09-14 [1]
##  stargazer              5.2.2     2018-05-30 [1]
##  stringi                1.6.2     2021-05-17 [1]
##  stringr              * 1.4.0     2019-02-10 [1]
##  SummarizedExperiment * 1.22.0    2021-05-19 [1]
##  survival               3.2-11    2021-04-26 [1]
##  systemfonts            1.0.2     2021-05-11 [1]
##  tibble               * 3.1.2     2021-05-16 [1]
##  tidygraph              1.2.0     2020-05-12 [1]
##  tidyr                * 1.1.3     2021-03-03 [1]
##  tidyselect             1.1.1     2021-04-30 [1]
##  tidytree               0.3.4     2021-05-22 [1]
##  tidyverse            * 1.3.1     2021-04-15 [1]
##  treeio                 1.16.1    2021-05-23 [1]
##  tweenr                 1.0.2     2021-03-23 [1]
##  units                  0.7-2     2021-06-08 [1]
##  utf8                   1.2.1     2021-03-12 [1]
##  uuid                   0.1-4     2020-02-26 [1]
##  vctrs                  0.3.8     2021-04-29 [1]
##  vegan                * 2.5-7     2020-11-28 [1]
##  viridis                0.6.1     2021-05-11 [1]
##  viridisLite            0.4.0     2021-04-13 [1]
##  withr                  2.4.2     2021-04-18 [1]
##  xfun                   0.24      2021-06-15 [1]
##  XML                    3.99-0.6  2021-03-16 [1]
##  xml2                   1.3.2     2020-04-23 [1]
##  xtable                 1.8-4     2019-04-21 [1]
##  XVector                0.32.0    2021-05-19 [1]
##  yaml                   2.2.1     2020-02-01 [1]
##  zip                    2.2.0     2021-05-31 [1]
##  zlibbioc               1.38.0    2021-05-19 [1]
##  source                               
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  Github (tomwenseleers/export@1afc8e2)
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  Bioconductor                         
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
##  CRAN (R 4.1.0)                       
##  CRAN (R 4.1.0)                       
##  Bioconductor                         
## 
## [1] D:/R_LIBS_USER
## [2] D:/Program Files/R/R-4.1.0/library</code></pre>
</div>
<div id="reference" class="section level2 unnumbered">
<h2>Reference</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-herbergDiagnosticTestAccuracy2016" class="csl-entry">
Herberg, Jethro A., Myrsini Kaforou, Victoria J. Wright, Hannah Shailes, Hariklia Eleftherohorinou, Clive J. Hoggart, Miriam Cebey-López, et al. 2016. <span>“Diagnostic Test Accuracy of a 2-Transcript Host RNA Signature for Discriminating Bacterial Vs Viral Infection in Febrile Children.”</span> <em>JAMA</em> 316 (8): 835. <a href="https://doi.org/10.1001/jama.2016.11236">https://doi.org/10.1001/jama.2016.11236</a>.
</div>
<div id="ref-loveModeratedEstimationFold2014" class="csl-entry">
Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. <span>“Moderated Estimation of Fold Change and Dispersion for RNA-Seq Data with DESeq2.”</span> <em>Genome Biology</em> 15 (12). <a href="https://doi.org/10.1186/s13059-014-0550-8">https://doi.org/10.1186/s13059-014-0550-8</a>.
</div>
<div id="ref-subramanianGeneSetEnrichment2005" class="csl-entry">
Subramanian, Aravind, Pablo Tamayo, Vamsi K. Mootha, Sayan Mukherjee, Benjamin L. Ebert, Michael A. Gillette, Amanda Paulovich, et al. 2005. <span>“Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles.”</span> <em>Proceedings of the National Academy of Sciences of the United States of America</em> 102 (43): 15545–50. <a href="https://doi.org/10.1073/pnas.0506580102">https://doi.org/10.1073/pnas.0506580102</a>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
